Index: node-16.17.0/examples/axios-example.cc
===================================================================
--- node-16.17.0.orig/examples/axios-example.cc
+++ node-16.17.0/examples/axios-example.cc
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #define NAPI_EXPERIMENTAL
+#define NAPI_EMBEDDING
 #include <napi.h>
 
 int main() {
Index: node-16.17.0/examples/debugger-example.cc
===================================================================
--- node-16.17.0.orig/examples/debugger-example.cc
+++ node-16.17.0/examples/debugger-example.cc
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #define NAPI_EXPERIMENTAL
+#define NAPI_EMBEDDING
 #include <napi.h>
 
 int main(int argc, char *argv[]) {
Index: node-16.17.0/examples/libnode-napi-example-c.c
===================================================================
--- node-16.17.0.orig/examples/libnode-napi-example-c.c
+++ node-16.17.0/examples/libnode-napi-example-c.c
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <string.h>
 #define NAPI_EXPERIMENTAL
+#define NAPI_EMBEDDING
 #include <node_api.h>
 
 int main() {
