Index: node-18.8.0/test/fixtures/keys/Makefile
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/Makefile
+++ node-18.8.0/test/fixtures/keys/Makefile
@@ -111,7 +111,7 @@ ca2-cert.pem: ca2.cnf
 # ('password' is used for the CA password.)
 #
 ca3-key.pem:
-	openssl genrsa -out ca3-key.pem 1024
+	openssl genrsa -out ca3-key.pem 4096
 
 ca3-csr.pem: ca3.cnf ca3-key.pem
 	openssl req -new \
@@ -137,7 +137,7 @@ ca3-cert.pem: ca3-csr.pem ca3-key.pem ca
 # ('password' is used for the CA password.)
 #
 ca4-key.pem:
-	openssl genrsa -out ca4-key.pem 1024
+	openssl genrsa -out ca4-key.pem 4096
 
 ca4-csr.pem: ca4.cnf ca4-key.pem
 	openssl req -new \
@@ -240,7 +240,7 @@ fake-startcom-root-cert.pem: fake-startc
 #
 
 agent1-key.pem:
-	openssl genrsa -out agent1-key.pem 1024
+	openssl genrsa -out agent1-key.pem 4096
 
 agent1-csr.pem: agent1.cnf agent1-key.pem
 	openssl req -new -config agent1.cnf -key agent1-key.pem -out agent1-csr.pem
@@ -275,7 +275,7 @@ agent1-verify: agent1-cert.pem ca1-cert.
 #
 # Generate new private key
 agent2-key.pem:
-	openssl genrsa -out agent2-key.pem 1024
+	openssl genrsa -out agent2-key.pem 4096
 
 # Create a Certificate Signing Request for the key
 agent2-csr.pem: agent2-key.pem agent2.cnf
@@ -297,7 +297,7 @@ agent2-verify: agent2-cert.pem
 #
 
 agent3-key.pem:
-	openssl genrsa -out agent3-key.pem 1024
+	openssl genrsa -out agent3-key.pem 4096
 
 agent3-csr.pem: agent3.cnf agent3-key.pem
 	openssl req -new -config agent3.cnf -key agent3-key.pem -out agent3-csr.pem
@@ -321,7 +321,7 @@ agent3-verify: agent3-cert.pem ca2-cert.
 #
 
 agent4-key.pem:
-	openssl genrsa -out agent4-key.pem 1024
+	openssl genrsa -out agent4-key.pem 4096
 
 agent4-csr.pem: agent4.cnf agent4-key.pem
 	openssl req -new -config agent4.cnf -key agent4-key.pem -out agent4-csr.pem
@@ -363,7 +363,7 @@ ca2-crl.pem: ca2-key.pem ca2-cert.pem ca
 #
 
 agent5-key.pem:
-	openssl genrsa -out agent5-key.pem 1024
+	openssl genrsa -out agent5-key.pem 4096
 
 agent5-csr.pem: agent5.cnf agent5-key.pem
 	openssl req -new -config agent5.cnf -key agent5-key.pem -out agent5-csr.pem
@@ -388,7 +388,7 @@ agent5-verify: agent5-cert.pem ca2-cert.
 #
 
 agent6-key.pem:
-	openssl genrsa -out agent6-key.pem 1024
+	openssl genrsa -out agent6-key.pem 4096
 
 agent6-csr.pem: agent6.cnf agent6-key.pem
 	openssl req -new -config agent6.cnf -key agent6-key.pem -out agent6-csr.pem
@@ -499,7 +499,7 @@ agent9-cert.pem: agent9-csr.pem
 #
 
 agent10-key.pem:
-	openssl genrsa -out agent10-key.pem 1024
+	openssl genrsa -out agent10-key.pem 4096
 
 agent10-csr.pem: agent10.cnf agent10-key.pem
 	openssl req -new -config agent10.cnf -key agent10-key.pem -out agent10-csr.pem
@@ -627,7 +627,7 @@ dsa_public_1025.pem:
 	openssl dsa -in dsa_private_1025.pem -pubout -out dsa_public_1025.pem
 
 rsa_private.pem:
-	openssl genrsa -out rsa_private.pem 2048
+	openssl genrsa -out rsa_private.pem 4096
 
 rsa_private_encrypted.pem: rsa_private.pem
 	openssl rsa -aes256 -in rsa_private.pem -passout 'pass:password' -out rsa_private_encrypted.pem
@@ -657,7 +657,7 @@ rsa_public_sha1_signature_signedby_rsa_p
 	openssl dgst -sha1 -sign rsa_private_pkcs8.pem -out rsa_public_sha1_signature_signedby_rsa_private_pkcs8.sha1 rsa_public.pem
 
 rsa_private_b.pem:
-	openssl genrsa -out rsa_private_b.pem 2048
+	openssl genrsa -out rsa_private_b.pem 4096
 
 I_AM_THE_WALRUS_sha256_signature_signedby_rsa_private_b.sha256: rsa_private_b.pem
 	echo -n "I AM THE WALRUS" | openssl dgst -sha256 -sign rsa_private_b.pem -out I_AM_THE_WALRUS_sha256_signature_signedby_rsa_private_b.sha256
Index: node-18.8.0/test/fixtures/keys/agent1-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent1-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent1-cert.pem
@@ -1,18 +1,26 @@
 -----BEGIN CERTIFICATE-----
-MIIC2DCCAkGgAwIBAgIJAOzJuFYnDamoMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
+MIIEXDCCA8WgAwIBAgIJAOzJuFYnDampMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
 BAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoMBkpveWVu
 dDEQMA4GA1UECwwHTm9kZS5qczEMMAoGA1UEAwwDY2ExMSAwHgYJKoZIhvcNAQkB
-FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0xODExMTYxODQyMjFaGA8yMjkyMDgzMDE4
-NDIyMVowfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQHDAJTRjEP
+FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0yMjA5MDMxMTA5MDlaGA8yMjk2MDYxNzEx
+MDkwOVowfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQHDAJTRjEP
 MA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQDDAZhZ2Vu
-dDExIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0GCSqGSIb3
-DQEBAQUAA4GNADCBiQKBgQDvVEBwFjfiirsDjlZB+CjYNMNCqdJe27hqK/b72AnL
-jgN6mLcXCOABJC5N61TGFkiF9Zndh6IyFXRZVb4gQX4zxNDRuAydo95BmiYHGV0v
-t1ZXsLv7XrfQu6USLRtpZMe1cNULjsAB7raN+1hEN1CPMSmSjWc7MKPgv09QYJ5j
-cQIDAQABo2EwXzBdBggrBgEFBQcBAQRRME8wIwYIKwYBBQUHMAGGF2h0dHA6Ly9v
-Y3NwLm5vZGVqcy5vcmcvMCgGCCsGAQUFBzAChhxodHRwOi8vY2Eubm9kZWpzLm9y
-Zy9jYS5jZXJ0MA0GCSqGSIb3DQEBCwUAA4GBAHrKvx2Z4fsF7b3VRgiIbdbFCfxY
-ICvoJ0+BObYPjqIZZm9+/5c36SpzKzGO9CN9qUEj3KxPmijnb+Zjsm1CSCrG1m04
-C73+AjAIPnQ+eWZnF1K4L2kuEDTpv8nQzYKYiGxsmW58PSMeAq1TmaFwtSW3TxHX
-7ROnqBX0uXQlOo1m
+dDExIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjANBgkqhkiG
+9w0BAQEFAAOCAg8AMIICCgKCAgEAk8XfZdGOl4FHdm9oxnL090RiXRryrEMbEFAH
+ruQNGxfJw4oHJQkV7v55Xkp7GWCcqHukV2mkKuVB2kI2SiTHN96hiSVg4gMAmUgO
+8mlh+kYHuhnqRiIxugAt1XAnlBtNox1IlBh36tGdIeMbHWPK1cA8qYXpMwPtCnC3
+oEPP3+sqxNQb1s4/hIgF2gjcp3Pb2V29cNmO8TjgokVKFPmBIML1PSXQcNvVDk2t
+z09ASF55MMPHL1Yai//9VJ8LgZmN3yn8eBsjQfF8zUHOGfR3d6Ta5ywIXqXB2cvL
+CIaRN9hrX8tDFe+mOD0ctoqPZJ9J0+JqiYh0VqzReuz8dg8jkVdDxbdE3ZhMm3hr
+eBVU0lomCAKnsc6S6Wk9u+LvyAzyV5YDslqoP1obgdjpPJm5SdQ3ula3XPSmfvJJ
+nn3XR/cabPKy9DzVLTbh8QaM0bxX38ba8CLYMJLE4eOHLiSUCKkrcn/Vyrmq1p0Y
+AssCcGhNPcgC4YwKI6jxEjYJUUICV7EyXY3aF8d53dGi355Xc5T/+q5caeMRBq81
+BfMjhQOCegJpp/PumN+XFZhaKciVsFkG6j3p1Cuf26HTtGrlgoxvL5vZBoqUG5v2
+W+Z3GuAbN9uXOBJ6OjTsZglTszaYG6ArTTu+l0AezoFnUZnkeJ3UTiK/Bo5vUNwi
+xVfVSI0CAwEAAaNhMF8wXQYIKwYBBQUHAQEEUTBPMCMGCCsGAQUFBzABhhdodHRw
+Oi8vb2NzcC5ub2RlanMub3JnLzAoBggrBgEFBQcwAoYcaHR0cDovL2NhLm5vZGVq
+cy5vcmcvY2EuY2VydDANBgkqhkiG9w0BAQsFAAOBgQCRSYfYjiaQ/sndWHt9ns4C
+eTrO9rsxw0r8rKVL7xjEbe/JbV+nos0mm0Xl79lYLqZ47/TRrkcqWc636QVulJk5
+CD+woBzB0n6N+HTwNDgzF4qGePxYduk2kdGgyvh82wt+atfSm+6m5Xhus9r6nyaB
+R0Lxyx+v43AvdKPrnFArwg==
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent1-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent1-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent1-csr.pem
@@ -1,13 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB4jCCAUsCAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
+MIIE5zCCAs8CAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQD
-DAZhZ2VudDExIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0G
-CSqGSIb3DQEBAQUAA4GNADCBiQKBgQDvVEBwFjfiirsDjlZB+CjYNMNCqdJe27hq
-K/b72AnLjgN6mLcXCOABJC5N61TGFkiF9Zndh6IyFXRZVb4gQX4zxNDRuAydo95B
-miYHGV0vt1ZXsLv7XrfQu6USLRtpZMe1cNULjsAB7raN+1hEN1CPMSmSjWc7MKPg
-v09QYJ5jcQIDAQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3
-b3JkMA0GCSqGSIb3DQEBCwUAA4GBAN3UIAdShj7eA91fH8m8UQBJndgigNwt88qk
-S2kS3XfZqkEawMu2HF/y5yWX7EyGs7OkRXZxJSR67GlgdrTi82qCBC3H2xF7fKXr
-s5b6ges5NZFjEA9JTvX5PFSAfo5APbXuuhRWBdxvagi00szTnYiaKgGU4C/dZWAz
-E0/tTFT4
+DAZhZ2VudDExIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAk8XfZdGOl4FHdm9oxnL090RiXRry
+rEMbEFAHruQNGxfJw4oHJQkV7v55Xkp7GWCcqHukV2mkKuVB2kI2SiTHN96hiSVg
+4gMAmUgO8mlh+kYHuhnqRiIxugAt1XAnlBtNox1IlBh36tGdIeMbHWPK1cA8qYXp
+MwPtCnC3oEPP3+sqxNQb1s4/hIgF2gjcp3Pb2V29cNmO8TjgokVKFPmBIML1PSXQ
+cNvVDk2tz09ASF55MMPHL1Yai//9VJ8LgZmN3yn8eBsjQfF8zUHOGfR3d6Ta5ywI
+XqXB2cvLCIaRN9hrX8tDFe+mOD0ctoqPZJ9J0+JqiYh0VqzReuz8dg8jkVdDxbdE
+3ZhMm3hreBVU0lomCAKnsc6S6Wk9u+LvyAzyV5YDslqoP1obgdjpPJm5SdQ3ula3
+XPSmfvJJnn3XR/cabPKy9DzVLTbh8QaM0bxX38ba8CLYMJLE4eOHLiSUCKkrcn/V
+yrmq1p0YAssCcGhNPcgC4YwKI6jxEjYJUUICV7EyXY3aF8d53dGi355Xc5T/+q5c
+aeMRBq81BfMjhQOCegJpp/PumN+XFZhaKciVsFkG6j3p1Cuf26HTtGrlgoxvL5vZ
+BoqUG5v2W+Z3GuAbN9uXOBJ6OjTsZglTszaYG6ArTTu+l0AezoFnUZnkeJ3UTiK/
+Bo5vUNwixVfVSI0CAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAgEAFVK1rBAhSVMxX6mfP4XKbJD+tLgR
+V1MDDrfJny11T+hmJ6H9QK6BfaPCtFFaWA+N1IHOzmvJ+izlpn0yHk9x5QCCk/H+
+brMOqHXL90jZu1kG2JYzL5plv1MzMfyd1gS4ZJyrLp49ZISCGHVL1NatEKSrAXHY
+bMxQ0byQEJhFTR/1LHPAU8PS5k2UYvF0IQoVYNIZ/vgYQ+fKNpxYvC214od9W0hz
+yeQmBSrqcxphRel1hoW89CXggGPWlrzzK6cbhqNiuiNJR4sS6SFn0vB14QcqC7/l
+PK5BltDN/OO2io2Dt80/o/yKAx+zbOCnM84pp9SOeh948r85y7uWY0UKI4N7fCo5
+jGpoAJAnav1+kT4riHRf+3rw/KFFk/pO73eH77LPeEy+8F/IPn3Cv6j4mDpXfpCS
+lZH0wN1HfFHhmjOoXm7Sboveqi3rSiT4svBQ63/eiWr8vN9fIHG8lwSlifOKNxdY
+QCkkzwPbxcZ562Aai1HiKMnr8X29WTZfBoXkmVrnSa1xvfstMSsIjnuKhew4aFm6
+UkZsrffNzqik2apx9YJjeSNJC5Lpunyc9TLRfFp2T+bMDfKGfRLKehreJrGB8Kc9
+O+zzNP37xm97cSiRsy0tr1Mi70TJwt+Nudg+JW9HfSE87xXgiaqXgAXEPphlB7ZO
+5go+ruyNJVw6FoE=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent1-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent1-key.pem
+++ node-18.8.0/test/fixtures/keys/agent1-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXQIBAAKBgQDvVEBwFjfiirsDjlZB+CjYNMNCqdJe27hqK/b72AnLjgN6mLcX
-COABJC5N61TGFkiF9Zndh6IyFXRZVb4gQX4zxNDRuAydo95BmiYHGV0vt1ZXsLv7
-XrfQu6USLRtpZMe1cNULjsAB7raN+1hEN1CPMSmSjWc7MKPgv09QYJ5jcQIDAQAB
-AoGAbqk3TlyHpKFfDarf6Yr0X9wtuQJK+n+ACt+fSR3AkbVtmF9KsUTyRrTTEEZT
-IXCmQgKpDYysi5nt/WyvB70gu6xGYbT6PzZaf1RmcpWd1pLcdyBOppY6y7nTMZA3
-BVFfmIPSmAvtCuzZwQFFnNoKH3d6cqna+ZQJ0zvCLCSLcw0CQQD6tswNlhCIfguh
-tvhw7hJB5vZPWWEzyTQl8nVdY6SbxAT8FTx0UjxsKgOiJFzAGAVoCi40oRKIHhrw
-pKwHsEqTAkEA9GABbi2xqAmhPn66e0AiU8t2uv69PISBSt2tXbUAburJFj+4rYZW
-71QIbSKEYceveb7wm0NP+adgZqJlxn7oawJBAOjfK4+fCIJPWWx+8Cqs5yZxae1w
-HrokNBzfJSZ2bCoGm36uFvYQgHETYUaUsdX3OeZWNm7KAdWO6QUGX4fQtqMCQGXv
-OgmEY+utAKZ55D2PFgKQB1me8r6wouHgr/U7kA+0Peba86TmOZMhIVaspD3JNqf4
-/pI1NMH1kF+fdAalXzsCQQCelwr9I3FWhx336CWrfAY20xbiMOWMyAhrjVrexgUD
-53Y6AhSaRC725pZTgO2PQ4AjkGLIP61sZKgTrXS85KmJ
+MIIJKQIBAAKCAgEAk8XfZdGOl4FHdm9oxnL090RiXRryrEMbEFAHruQNGxfJw4oH
+JQkV7v55Xkp7GWCcqHukV2mkKuVB2kI2SiTHN96hiSVg4gMAmUgO8mlh+kYHuhnq
+RiIxugAt1XAnlBtNox1IlBh36tGdIeMbHWPK1cA8qYXpMwPtCnC3oEPP3+sqxNQb
+1s4/hIgF2gjcp3Pb2V29cNmO8TjgokVKFPmBIML1PSXQcNvVDk2tz09ASF55MMPH
+L1Yai//9VJ8LgZmN3yn8eBsjQfF8zUHOGfR3d6Ta5ywIXqXB2cvLCIaRN9hrX8tD
+Fe+mOD0ctoqPZJ9J0+JqiYh0VqzReuz8dg8jkVdDxbdE3ZhMm3hreBVU0lomCAKn
+sc6S6Wk9u+LvyAzyV5YDslqoP1obgdjpPJm5SdQ3ula3XPSmfvJJnn3XR/cabPKy
+9DzVLTbh8QaM0bxX38ba8CLYMJLE4eOHLiSUCKkrcn/Vyrmq1p0YAssCcGhNPcgC
+4YwKI6jxEjYJUUICV7EyXY3aF8d53dGi355Xc5T/+q5caeMRBq81BfMjhQOCegJp
+p/PumN+XFZhaKciVsFkG6j3p1Cuf26HTtGrlgoxvL5vZBoqUG5v2W+Z3GuAbN9uX
+OBJ6OjTsZglTszaYG6ArTTu+l0AezoFnUZnkeJ3UTiK/Bo5vUNwixVfVSI0CAwEA
+AQKCAgBMuxxrFEtEZD6lQuhW025bzsh8q/eCdz3gVI4IxCYCAXIk9NEM7BDVjpGa
+4nXVEZOG8Ny+/v3dwDlXibYC/eHsItAiFKSf9wsijIYmXY4glKiCe95oGG1Vta4W
+/Cuw/hRsNFJxg9sQtVMwO2ri8eRq5W0aYYxbjxhQqYLP1yZjkQseNT1XZrUYrkDL
+nHpnAQQRdFlf9bgBjZUm7W+wQHZKerE41R/M2nlg+fjcgrxQV636Ik6VR9J1C7GR
+dRxWaVN/o5+TBhRis3AobJkGZSh7mcCI9RZLfyqPWwDMJoPVSa1HipYFF/FplmII
+bx5SE1wUh6bmT1KJhycm+1OEQgwnK/mB7Q+VRORuCgMDGkClkIykmCD7JVtqlvmy
++E+U2AdgNuMgNNVPCODqvnUi49eu81ITaWZAjbZwgwO+SusaZ/JG+gbKR5NuOpg5
+Uj6rbk+jaWB5R4ZFutAaWcBSqRPVGQp0TgyvSc8uUjrk7aOYHRonWAP9068PoF7o
+Yc+6j6mbWQr7/IdE1Sf/fmc/9SbeG5iULy2+QNSjkiFHOq7THCDTvoqOn6Qa2ZBp
+Qmm4qpGveVfc56DdofJfOUQChJgImnGF0TofKQ83JNO3dvDsO9ODXGIQox6/Frka
+WErHhc0abWBfucRYhZQ6lSgVdm0peWUTkWn43Z5ZcHqQqxuDCQKCAQEAwpPU5l5L
+sT63l55gI0k8pl91B6Z2xRmfREKQJy3qhYEl7gajUoL6uBenvQKoAoQcmvnv5LdG
+hNStXhhH+zM+0q8OP26Ufjz+HOCKcfLoQ8XVgHqFejB7dTQtEUPBYpN8N4m4G6ZD
+zjCT4rSUQnfsKlTqglrduKDUCB7siBz/cbF63R2tjaNuvlaBhPzeuw2ECiV6AX7Q
+377BcWDD0aSiDgYC6MNK+t1duAhMXs1LEED07KCG7FLO3G6BWmLbKnzDYBk4ZpnH
+Z3tWARXk6gezvetpf1kFwaRmw1x5ZBACGnH1jXgPdsNlD0xh4xvkqDpJ7HRCYn+U
+bS2DkAWO0J+23wKCAQEAwmuuBkk0wIKEJ7AXpm2feQ0k6p+XKInioM3HYSn08WBP
+n/0+/1FG/l1vnw273gH8WR17hTN0GHkEpcdwAe4+1PlJoluOkEGg4DZNLhiRXzXu
+8c50cDzYNWYO+bFUV+mK6u6tCsroSni9dYQKs796kALp+2lZ9sq0wOyPvr5DKfiV
+MRMpADQ+Jh/pBHROKKSAHJCCogxkxUB0GkUmX0HZLBJo/gJIX8NC58zfcN2JPn1P
+RK9IIp6URBqrkqjGSiePeadnvjjhI4ALAoELMd/BEfg0MHrSirwzqfOT/JhqKnPf
+T2r/02ynAUHhaBDCIMhhKp+EDS3e2iSD7dLZaeoKEwKCAQEAk90ZvhLMfJ5HkE3H
+iykdkOUbnzuuKCxHbu0SdOOcw+WX+GFM/egRNZKsYqNBH+BbiJ4E4o8wZMLmGIpl
+MQ0JYbusHSIgcCpThYhRTTSlstMFY0Q4aNmtCC0lTmhp7kpb7CyZH1HjVy5tQ6us
+uzI84xJyv6rjsWlul9lAMFOw/yo7Tws9l1QqVTSR6ojrjcRAoiFGrRo9CrsAb/Nz
++72bdQK2dUHrmeZdUPvlZWmm8GwcdDGtNqvbDYJghFp5aLk1E7M2vCmCci409lA4
+DGzb96iNHDx/dl9DxJRin1rW7lfJnb8uKTdldhG6lYbd5cWd5bWXuRhwp+MdhnXi
+yMGGiwKCAQB39yenm05gp2hScPb269emQzUPTTBhhEuB+9fWHaSVuF8JkqNnucg7
+tn3kv/23bLRAbM/GDkzJuQrkHF8B4feFRNlpxGMAom5ZS0hudBxniokJTeJNS04C
+DRqaehBYOrRMaL//09pn8X6+X56W8UT4TcQBLS2k+cl07cN46D/bSxcL9A2toY7j
+yXboF7gOngqrFYR1533atP2vJBIcXy5ZYnbEnbX42SyUibcqFpjUJVBfSmURb5Z/
+O5G46kmIgkamw2RhXQmW9TBV+DhnKvEgXG4fp/n4P08FxFYBXC7OSTd28sgutnGK
+dhNXOlf/EIXcXheGZlKHelapzhv3ncFfAoIBAQCsjdIhrPppsWxChMO/L8CP2qvj
+nNu95Z9T+7SADUrIM+PqHP0N1U5AEvVs1uwWhR4elZXKlQpV60yfOiWkVmn2ByIe
+znr3OFwAWXz60WkDcD9kDslQAU3+Xiwt3vy//gaPqngZw6iIv1SvxpPVQ4X1P1HY
+JvE03Yliv6+7m7gG2SoSJecGIT6Z472MM9AYByWMVQz3KS9iYHBZGAVRsR4tJl5U
+gNnsiLBEJTFPRCI8NCen64EEspArthlwm3PSc2gNbAKZKdxBSAapIP9n3xndy4T0
+xTG9UzTnJSb7nO/4DavGUHgjHEKMBaQUHUGF8DtDlnBgXS6Kz/hbYDmEAE3a
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent10-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent10-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent10-cert.pem
@@ -1,18 +1,26 @@
 -----BEGIN CERTIFICATE-----
-MIICfzCCAeigAwIBAgIJAOyvM6GMZDW6MA0GCSqGSIb3DQEBCwUAMIGIMQswCQYD
+MIIEAzCCA2ygAwIBAgIJAOyvM6GMZDW7MA0GCSqGSIb3DQEBCwUAMIGIMQswCQYD
 VQQGEwJVUzELMAkGA1UECAwCQ0ExCzAJBgNVBAcMAlNGMR8wHQYDVQQKDBZUaGUg
 Tm9kZS5qcyBGb3VuZGF0aW9uMRAwDgYDVQQLDAdOb2RlLmpzMQwwCgYDVQQDDANj
-YTQxHjAcBgkqhkiG9w0BCQEWD2NhNEBleGFtcGxlLm9yZzAgFw0xODExMTYxODQy
-MjFaGA8yMjkyMDgzMDE4NDIyMVoweDELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNB
+YTQxHjAcBgkqhkiG9w0BCQEWD2NhNEBleGFtcGxlLm9yZzAgFw0yMjA5MDMxMTA5
+MTRaGA8yMjk2MDYxNzExMDkxNFoweDELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNB
 MQswCQYDVQQHDAJTRjEfMB0GA1UECgwWVGhlIE5vZGUuanMgRm91bmRhdGlvbjEQ
-MA4GA1UECwwHTm9kZS5qczEcMBoGA1UEAwwTYWdlbnQxMC5leGFtcGxlLmNvbTCB
-nzANBgkqhkiG9w0BAQEFAAOBjQAwgYkCgYEArV2diVumrKDS5k81MrcdECnYYVZ5
-feQ/FZDqwEHM/zlXvs6vphU3rGmZeASMQEdHg7vUjzzvE8PDqJuJXKrC5lEO1OUY
-eUDhaZ/QvYS9tDp7qTJzORxT9im65dQH0Xq5JQwTy30hidQHxOgAkILNive07/Jk
-N1vle6TnZX6K/dkCAwEAATANBgkqhkiG9w0BAQsFAAOBgQAAg+FpvhA6coalWxGR
-acWiUbc7CJ4RWjlSeA+fhd1G00x0Hl5hjt6IAqEHe4T9fV41U05X1eo5KaN3jXWU
-IS56SVX8BxOhU53lr0iID0MpxMqttA9LgjE3fc6uAjThnx1zX50VGR4P8LQqG+HL
-WJUW0+3oJrOgRbJ6wAEs0iCcTg==
+MA4GA1UECwwHTm9kZS5qczEcMBoGA1UEAwwTYWdlbnQxMC5leGFtcGxlLmNvbTCC
+AiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAOTCvZyId7XQL8eGVE/+JKfJ
+XGNtzdLt5uxV3FBMFxP6oKtfKvwiggaw8LEU3TTeNx+eAwRmfgjoD2U98UYCXNwo
+O/Mzgg2dZpKer+Dmz39XyEs0wTzm58FspPo4ZcdafvHUrrZMwQ7MyKxDKSReAkT6
+Yng7XDL1hrRdhsCNXDorczhN843vhBx9vcrsCpNebAPG5uJJtorevirFIZnzMfj3
+zLc+EpsAJiPe2+57uvYs264//egoxCJ2XdSyyWkWIJf7FzMVVs0CboxKwLIBtqQI
+IpAvDH4fYL06carQwqyBLnA08DpN7dJ1x7toZ8fbu8tlnnGFeq6toZ7wfvpPew9O
+WRoxESHyjm6aBAvrAnPmbsGhNB++qhs/nvS05dnfMNtI+breo4Pr+ZXdfdEZ5qVN
+SgdhAZ5UuSGU22kmkfjtKvcCnr4DlQYqv0HFqdlENV6AXqmUW+b0GktRheOXd542
+YQwsdbvV3ZBLmBcSG2abTeg8e1zr3CDpGZU9oa2a1DyWHQI8F/6TxJG7u3gaLITt
+Ns37O2uf0ZlJpVIhP90g/U/Ewcg78/dtntAHVYevIFLdAVDMXrM5ztgnQ3GrPG3I
+GJirik71WQX5l+s547USvoBUED76IJKPigaGGpg9LkO90SUMD1qCEbws6271PyeY
+DIrtRumanoqqSanEH877AgMBAAEwDQYJKoZIhvcNAQELBQADgYEAdzHr5RgmR/uk
+QmQRn+6dTzKBiiAVqqmQsuqe87WHZ2KVvjUNrw0y1CoHAM81glxhr9qcuo86lIPR
+PdMtsrJecLSOCwYMPXMiAWt+u3sHxo0DZl3bRt8dQr1Nj8SWtuTDHu5JVDG0Oi69
+XT8ZWQVfE6dup82sOdKJQYl4On/hGHA=
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
 MIICkzCCAfygAwIBAgIJAJHwBmNgafKbMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
Index: node-18.8.0/test/fixtures/keys/agent10-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent10-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent10-csr.pem
@@ -1,13 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB3TCCAUYCAQAweDELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
+MIIE4jCCAsoCAQAweDELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEfMB0GA1UECgwWVGhlIE5vZGUuanMgRm91bmRhdGlvbjEQMA4GA1UECwwH
-Tm9kZS5qczEcMBoGA1UEAwwTYWdlbnQxMC5leGFtcGxlLmNvbTCBnzANBgkqhkiG
-9w0BAQEFAAOBjQAwgYkCgYEArV2diVumrKDS5k81MrcdECnYYVZ5feQ/FZDqwEHM
-/zlXvs6vphU3rGmZeASMQEdHg7vUjzzvE8PDqJuJXKrC5lEO1OUYeUDhaZ/QvYS9
-tDp7qTJzORxT9im65dQH0Xq5JQwTy30hidQHxOgAkILNive07/JkN1vle6TnZX6K
-/dkCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBwYXNzd29yZDAN
-BgkqhkiG9w0BAQsFAAOBgQBeyxGhHnFF0ifHhWUbqqMM9zJ5OhLjGsQ0gvmK/LHL
-vmGJ43XgeYiN/U6xREQ7DZMss+C14mfQvp5oM0zQRWwQhLgV7YlIIIe09CYTKTfC
-xxc18OJewNQUje5cG5aSMZb2HfHmLDaavAJqK0Yaoj69e+iEnAkVFVZALqlhezS+
-xQ==
+Tm9kZS5qczEcMBoGA1UEAwwTYWdlbnQxMC5leGFtcGxlLmNvbTCCAiIwDQYJKoZI
+hvcNAQEBBQADggIPADCCAgoCggIBAOTCvZyId7XQL8eGVE/+JKfJXGNtzdLt5uxV
+3FBMFxP6oKtfKvwiggaw8LEU3TTeNx+eAwRmfgjoD2U98UYCXNwoO/Mzgg2dZpKe
+r+Dmz39XyEs0wTzm58FspPo4ZcdafvHUrrZMwQ7MyKxDKSReAkT6Yng7XDL1hrRd
+hsCNXDorczhN843vhBx9vcrsCpNebAPG5uJJtorevirFIZnzMfj3zLc+EpsAJiPe
+2+57uvYs264//egoxCJ2XdSyyWkWIJf7FzMVVs0CboxKwLIBtqQIIpAvDH4fYL06
+carQwqyBLnA08DpN7dJ1x7toZ8fbu8tlnnGFeq6toZ7wfvpPew9OWRoxESHyjm6a
+BAvrAnPmbsGhNB++qhs/nvS05dnfMNtI+breo4Pr+ZXdfdEZ5qVNSgdhAZ5UuSGU
+22kmkfjtKvcCnr4DlQYqv0HFqdlENV6AXqmUW+b0GktRheOXd542YQwsdbvV3ZBL
+mBcSG2abTeg8e1zr3CDpGZU9oa2a1DyWHQI8F/6TxJG7u3gaLITtNs37O2uf0ZlJ
+pVIhP90g/U/Ewcg78/dtntAHVYevIFLdAVDMXrM5ztgnQ3GrPG3IGJirik71WQX5
+l+s547USvoBUED76IJKPigaGGpg9LkO90SUMD1qCEbws6271PyeYDIrtRumanoqq
+SanEH877AgMBAAGgJTAjBgkqhkiG9w0BCQcxFgwUQSBjaGFsbGVuZ2UgcGFzc3dv
+cmQwDQYJKoZIhvcNAQELBQADggIBAIzE/DhbSR9aiODaA66vPVVAjsf+yn1dmniI
+iRo7scheU5dBmYcJOqnkZ57QWbnB5qoYrcddJ17ijgA4fCV0IPnAunE6RCh4HBlx
+AtXcniC4ftZRnezQJdIUlR6j+w6NFQlYW8DR8uvROV13scoXngfC2dQCxrvsorTH
+6NlenyvMI1TH4wPTgqtFAaMie4HnP78J30fvTfzwhJisWiqwc6G7EtZ0cnpmJdH/
+0L8jgDkfetCzTBb+4JjSdElNR0rYZZYDK5QbVkhNw9K+vvB+UcAwANDkTVcwRzeZ
+gmujiXLouALHuaKjTpYkRa5M8TDVMHVNUJPphRCzQcDVGJMOCxS2jmHe7Q67uvWm
+vSuEsuCOjc5K5dz2La2d7q4GmzVb+3IdFqSJ3QfWnARMp2KttxpmYm4VbpNBj+Z0
+jNA+uWgOHOmpsOMn2u4A/72o5QNPM6jcdifiY2KALOT+uPcuOBLTIEBmN9s1/9Yr
+fdTDsI26IOMf6NfaaFyggW+mut1SUFQt5XDP3B5MLzEjr9/vi9aKl8pmGqHwI+LN
+UGl5l/e7I77e+FCbtuw8pmeolmJfK2T4EKi1hC1axz32zwTh5IkPirn6HdsEYOUb
+qyMIUFDI4Df4BxeFJtcApgyV8b2lko3QdiihckGJvyd0FWPG7utOFh9bffrJR7IF
+2rTgD8U+
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent10-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent10-key.pem
+++ node-18.8.0/test/fixtures/keys/agent10-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXAIBAAKBgQCtXZ2JW6asoNLmTzUytx0QKdhhVnl95D8VkOrAQcz/OVe+zq+m
-FTesaZl4BIxAR0eDu9SPPO8Tw8Oom4lcqsLmUQ7U5Rh5QOFpn9C9hL20OnupMnM5
-HFP2Kbrl1AfRerklDBPLfSGJ1AfE6ACQgs2K97Tv8mQ3W+V7pOdlfor92QIDAQAB
-AoGAIAjiaVVEMTXugpw0SlDH0ArLbwEZpgedGJEUr7348VhZPGrYzimxhexlbWX5
-vI7vSgpVNrqduts7tlY3RaZQKQzFkSqkUUb432bXkJLHNIspd0XHOO2Hy/ZbTg0n
-OIQes7C91Z/OLUi9esXoh4AMsAoxiHoVee0dkEJt8RoywNkCQQDk39QND1rQ2eJq
-Fcfj/v6fXgsHmQT16w2Ii9P5uPAeIGrGcrsCoVWrsh+wjSlYc7emGV8JINiltNhZ
-fSg6ux8/AkEAwemf5LryUDCoZ68MlAYMcH+G7gtm06d+FUFpBclT3hJeXnUAdlyU
-6kCvazcVTQQKDTWIS1oIBuleVmc/VWc05wJBAMVRZyq/QydtwTJ+hq+8pl5VIKMz
-PECbnjZLfrv7wh/nCMcAINRarVZyIbn/aVbVpM3xb6qaA82QxTkZmvZPXtcCQFqx
-pjMYrNSMrXcxDDT/Tzoeq0ES3BkKMZJHcZNfQnaPKMwM9RZm3s9hSapfrPrEdN8Q
-tppnlXGGHLVUvO54wukCQBKbjspQONRZFBh9Fdeyf/rX+inBLhuMSnsY3FeBp0c6
-TAPbyuiezn7axr9kILojdjZgXK1b+MTHSEjqCpHwIvg=
+MIIJKQIBAAKCAgEA5MK9nIh3tdAvx4ZUT/4kp8lcY23N0u3m7FXcUEwXE/qgq18q
+/CKCBrDwsRTdNN43H54DBGZ+COgPZT3xRgJc3Cg78zOCDZ1mkp6v4ObPf1fISzTB
+PObnwWyk+jhlx1p+8dSutkzBDszIrEMpJF4CRPpieDtcMvWGtF2GwI1cOitzOE3z
+je+EHH29yuwKk15sA8bm4km2it6+KsUhmfMx+PfMtz4SmwAmI97b7nu69izbrj/9
+6CjEInZd1LLJaRYgl/sXMxVWzQJujErAsgG2pAgikC8Mfh9gvTpxqtDCrIEucDTw
+Ok3t0nXHu2hnx9u7y2WecYV6rq2hnvB++k97D05ZGjERIfKObpoEC+sCc+ZuwaE0
+H76qGz+e9LTl2d8w20j5ut6jg+v5ld190RnmpU1KB2EBnlS5IZTbaSaR+O0q9wKe
+vgOVBiq/QcWp2UQ1XoBeqZRb5vQaS1GF45d3njZhDCx1u9XdkEuYFxIbZptN6Dx7
+XOvcIOkZlT2hrZrUPJYdAjwX/pPEkbu7eBoshO02zfs7a5/RmUmlUiE/3SD9T8TB
+yDvz922e0AdVh68gUt0BUMxesznO2CdDcas8bcgYmKuKTvVZBfmX6znjtRK+gFQQ
+Pvogko+KBoYamD0uQ73RJQwPWoIRvCzrbvU/J5gMiu1G6ZqeiqpJqcQfzvsCAwEA
+AQKCAgEAqb7vX9XLrne63H0K1wioPnCmQ4lY7ylk7v2PK24qucKBsmg9xzX78ecd
+0ui+c9poYOepKXy8DPPVaLSb8e0S5rJcOcyAtoWTdTyXGXVrhvymPVf6A6z7T4XE
+rX9n1pOKGvxMYC14G/lM9rYnMzsmx/RbWQgDFDUoLXIw6XR7eRaAusjs7udJyrJr
+QMYB2vR3hh3zePpln4pE5gu9MTVKYYnOuDIxMB7eLkPSQpZBVyZWJGiLidemIkRe
+v9OirH2k1eGVh2xQKnOhgCx/7BZlZjY7H24qAPRbnk5dxfh+W+7HfK9ringE/uQy
+vxdr2ttr8DOJbIc1T6S7hurB7g9IKd4j7xcZ+FdiTJEfVrPm2az4qYxHJsvUHlGL
+2l6Yn45CcSuUOckCjtqR4ti0w6wQm9Q35EQelMaN2n2Yf3hPW/vDtnwVppvlgGOK
+YDNjyZNCGycpyf8BaNrPEKX287B1LNtErY0GI+P59iun7m8Gt69n+y4BQJAxTFoA
+px89bg8WyHgoPqQ9fhBLgv0/mbfp3BTznDH4V0vZUl1WEeVuw12J6d5gzNd2lLc5
+UTScl+lPJCZm4W9b7C+ZLCXvUwONpF9WxYOWAzkWB+LL+nOWy8M7xxVNhQ3do5QK
+qpmWjygN13BPUEhT73CGWHvdQwZNSgJI7MQ0w56VItVjZyZZ3AECggEBAPfzrP2a
+omKSHNyCnePlA/B0pL5E0tJGoIgpAnG1KT72FEK9M2hryx9BA/G8QS6xoO2/4Hk2
+tfNw+ab59ZoYrDbUwAOhk9uMINIt/wLn8if2uzkU0PthObTRMVn+QYfo1ybX9xHL
+CzIUGnOEkFNSSCJbfKya+oEPVbj+k9PJiqAodTGFQ+SBavF2DWRLZqseu+soJwxz
+ftHX6Alb5Rj945e6DWhcd+ljfruvkQ3I/6v/o33LGB16SvtXgVZOPtSo21P+fbIE
+SvISBaMuwd/UHv/r/wK6xmF42R/PK4RJonCXPVrLtJhw2m0KE0tKJoyYJNIEPuy6
+ohJHK4+u17mnC/sCggEBAOwvmTcUP9umgy9fIQFjzSju3Z2qCdoQcMIq03mN1akS
+0yng5zOHSUwRz/5kwGtLjHBdmnn8k4584zvZcbEDiXy8r+1fwp8nROA7v/THPwSD
+4actSWfzgfFCaH5RSW9k2KQS3/fvV2ZX8QTM/CzFRbuFMdLralEENndeE5jBnABX
+Z+0fUIcMxI+cQKOxRBFc9Zo3aY3M4JR8kiAeufdrUN5LZp6VtO5o8lTk6lDPX7Wv
+ZsUZ28oEV6y5SnXjpIJzUC6eVq4JatlmI7LHIdY0xrcG4famHgtJHk0MPL3b/XXv
+b228O0uUA+LB704l/VqCz/G94vbNEpaaePrsXwoj2QECggEARSQzEgjjzT7XQ3o8
+0wbSoWO1aMVCUJaXP0DISLLuzHrMTYv64NxmKNH1apxHROENch62Vra16CuRwkkj
+fpPwud17qceNtoQNxu5ClSnesdwwewtd9ys2b0CCVYj79w0rqc5BPDn1ZkKK2bmW
+kVV5ffMrAeq5ffGrwsliqYUi3eRKQcPamcao7qovs87+0yrf1gQ9V1CBhbyCHNH+
+QYjSxUWBZZ+bZH51IZskhMaO3QSNL4/JqramYA4Xp9xr0ts/Rd3KaTCfQX9bCRUJ
+xQOenVjOpJE2romHakJX7SynWmQiHMZ/Fp8pmYxcsvr7YusuIU5DybmFZjloYUmt
+KDz3rwKCAQEAiniRlHg7xDSInwJf4Ge5xbDdn20sn3uaEEx5db+RL7IIuXnRMyWy
+9HVRA23gXkeJy+slATXy8MwJkXI3tLdFKSua3UmSWir0t55wOTuDl8sWRiRH8DkU
+yvTLPJu9uSBf3znts3fQPwF6Hf1sL3dtHbbGcklxKJSWtj8isCF6q7yTArOkS7lR
+mHzcbNwl8yQUHlk0el1Ood+y8D/zebqiS+HbBUjyLgC7ashXFA+wpl1ynEY/1N3c
+5TgxDpLrRUwEC4DrgteknFbEwTD8hpCknHPdkPTtkmJI8JUnG4zZugYpxyVfgj0C
+VjSrb57YPfOJpMpWQaoTptZj7YEda11/AQKCAQA9Y1ESJc+TSgUmxFAg6UuMU71U
+KPWRjClNKwMDvradz2qQ1Z1dnrnEZfTiSsFyXUs31Iah4QTvyV5MZIreuf91NJ0O
+oCkH0T7gksA9bYTwIMYZiqFL1hN/o11KuSJB0mY9fIVlrztNDo61II+lZEOJ4dyx
+tDDh/cQCItITiDbtYcPV0oGWgVbU8zReUKOSrWK1ldK561YxTTGrYd781zaUhf8Z
+CbMJKXREc0FZJId9XrffcgvpCYx9tBYwKj9xddb6ByvgZhYe9aJtKYPf08ElMAXM
+xn9gJgy76jVFff2/ybmkd/ZBeHFOnY+p1WyR3OtmNyQQqJjaVSBO2oWuCSP5
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent2-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent2-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent2-cert.pem
@@ -1,16 +1,32 @@
 -----BEGIN CERTIFICATE-----
-MIICczCCAdwCCQCtrfdcbYDS0jANBgkqhkiG9w0BAQsFADB9MQswCQYDVQQGEwJV
-UzELMAkGA1UECAwCQ0ExCzAJBgNVBAcMAlNGMQ8wDQYDVQQKDAZKb3llbnQxEDAO
-BgNVBAsMB05vZGUuanMxDzANBgNVBAMMBmFnZW50MjEgMB4GCSqGSIb3DQEJARYR
-cnlAdGlueWNsb3Vkcy5vcmcwIBcNMTgxMTE2MTg0MjIxWhgPMjI5MjA4MzAxODQy
-MjFaMH0xCzAJBgNVBAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzAN
-BgNVBAoMBkpveWVudDEQMA4GA1UECwwHTm9kZS5qczEPMA0GA1UEAwwGYWdlbnQy
-MSAwHgYJKoZIhvcNAQkBFhFyeUB0aW55Y2xvdWRzLm9yZzCBnzANBgkqhkiG9w0B
-AQEFAAOBjQAwgYkCgYEAq2BtovcwlIUNVAi3ukbyQveZNg3VkkxB6Ih+CL70VEX3
-0ji5EpQqGjwk5Ub6tRYfuGHwBWSO8+W2PwtdplDbuIOoK5MAp+wkDRCC8H1l4i8c
-dOA97Edw3nRU6MiPrUiirV1u2liHYp6YDLWix80UiNnH6EOLFd5Wy5s7CSt4rEkC
-AwEAATANBgkqhkiG9w0BAQsFAAOBgQCg75WmaDjpwBmMAl/+5oK8D7p2V98ChhyV
-bIwhpLPo9jf7iPW9VayxQKvV5HXuiT1vpIjuvhbRRH3wGj3by0TtI+sqECcPKu1v
-5bg2e+bX8s1OXFJh+x93KCnrcFNEwOVk9VjUX+ilJdTkdcGA75N4ZO7qEV5wKl9g
-658PRZl3KA==
+MIIFgzCCA2sCFFMncSqgKY+i2kq64Lu8zO26Z7hoMA0GCSqGSIb3DQEBCwUAMH0x
+CzAJBgNVBAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoM
+BkpveWVudDEQMA4GA1UECwwHTm9kZS5qczEPMA0GA1UEAwwGYWdlbnQyMSAwHgYJ
+KoZIhvcNAQkBFhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0yMjA5MDMxMTA5MTBaGA8y
+Mjk2MDYxNzExMDkxMFowfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYD
+VQQHDAJTRjEPMA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYD
+VQQDDAZhZ2VudDIxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIIC
+IjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAzFbysXv+9Krce+YF58ByJq4v
+rQ1/0t2jHVlcWmAVONDpG4ntF8y02xlkwmZGLIX0EoQAbSI7gvJT+s3GEc2WHdH9
+rTQu4vqGTRdec7bcBZryoW9mZiZyc5xEsXEWcFzZNhnrRbUPeJ7jsR8qAdAkDzRV
+W4uCKWjdbetVDGbsEdO5kzjaL2rmLeYq17x+obGuvSzuzV4lZgAujQgug7bapQjg
+RvVbwPi+kDFjVDlTgNoX8/xK2mGnrg3Eh5XAGDp95uqZifu0MGZ2MUEhjJTjw/7l
+UxdOnVXzEPf5FhuWHXx3Jmuy18pvDume+Pwww1r/R6+9w8k6qJaC/fgKlr3pVxqf
+83BkowsdeGaj24mdhz/rjv82vTdgimWr98O3tkiTUAX3J7T+Bc/SpFqLjB+rdki4
+yAyXkgxDQa9ZpMOz0y9+NJYP8r7VH24pk9YVABYI8MHyLVIYoVew7LE0uVDRp7a8
+tP0LClHOzemPgj6Z+HA9rSfqkh98txPpzhGJL0/j/XirY8vIp6vfkL+V8cYtGJ87
+lQvspe24ZDWbSjCv0L560mNu48VRLGk1HwhzcapzVIdb2lX+tJCPBnakK23YXFZX
+qFXlLNwE3dGAPprxyJyqWj4Brgkz8qXrFDJIOATnh5trV8Ogy3ytM1xDB4ohRtjv
+PlUVzPzim2UwVV4oJfECAwEAATANBgkqhkiG9w0BAQsFAAOCAgEAzD4rAd7agBgo
+WFKt5v3mUuBOaYJJ4jwd9wbvpd96rbsg17MLwa6U8XDLQh5j6gHLFDP1pAbdHeC3
+SPIj6PwsLdFCteN9yVoJd9SiNOjmf8DOQtSPCiczlOHQg3jsLh4Kj1FXimbESFvX
+imk+KNoR7DWX29Xrcqnqtnau2Eb86dZfSAvFCh9KKFCcXwHbhJEuiPjq9dzYqheT
+I7Vgr0piXy3LYWzeIjpgKzRg/tRr7PINqL/2mSS/yP9VGk88dLPaFk7R1NiXy2w/
+fx8QS0UOuOPOGIqGg1jF/AxRLHrlURF/daRFikidFpfsqMMaUn1BhDKT/yZaKG39
+x640t16ZLKeMcYDmswdx5tu+YiNckzYaZgrxmBk6NIvhemDbndCINRsbgJEKTEDo
+UspA/tWTFRWVUfW+zNBwx4VaOdhI3Xhj7tInMfnvJVvo5vPXIQ8m7WwdE9CYu4/H
+xxJ3pU7SehGhUF+tV/HCEBuudUKaVVXNr+Cn6Qqp+sCyk0Wb1iXC/alyus/C7WAa
+s58VLUNAPRsreV91p45qnCMwH8hMoVgx7AryUn1kKx5Ani9Kd7VPaEbVkdAWhqyD
+J6gXcoDBEy/GkFEl7H623wgjarKp/wqTh+m1AHvRtsBlTlZ6Tid7opdGB6v9Dx6b
+F7o9bZYIzydgFkjcMAPDNmkgGjd5eqE=
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent2-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent2-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent2-csr.pem
@@ -1,13 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB4jCCAUsCAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
+MIIE5zCCAs8CAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQD
-DAZhZ2VudDIxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0G
-CSqGSIb3DQEBAQUAA4GNADCBiQKBgQCrYG2i9zCUhQ1UCLe6RvJC95k2DdWSTEHo
-iH4IvvRURffSOLkSlCoaPCTlRvq1Fh+4YfAFZI7z5bY/C12mUNu4g6grkwCn7CQN
-EILwfWXiLxx04D3sR3DedFToyI+tSKKtXW7aWIdinpgMtaLHzRSI2cfoQ4sV3lbL
-mzsJK3isSQIDAQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3
-b3JkMA0GCSqGSIb3DQEBCwUAA4GBAJqJXQdhuPxsJA6O/yWt9t9lQIgoYCQyCG2w
-Xl0n84f14WDi/N9rF0IfGMSVWoLDCc5gcoqKal0X/vQI4lpPiZ0hctU5cXru1Pvi
-yfDbIPB0td7POf3Q3Ge3a3RHf4I4cfRuzA6jfzMlorpgQmAKL+sstC94LZZnDiNp
-ihciaeK7
+DAZhZ2VudDIxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAzFbysXv+9Krce+YF58ByJq4vrQ1/
+0t2jHVlcWmAVONDpG4ntF8y02xlkwmZGLIX0EoQAbSI7gvJT+s3GEc2WHdH9rTQu
+4vqGTRdec7bcBZryoW9mZiZyc5xEsXEWcFzZNhnrRbUPeJ7jsR8qAdAkDzRVW4uC
+KWjdbetVDGbsEdO5kzjaL2rmLeYq17x+obGuvSzuzV4lZgAujQgug7bapQjgRvVb
+wPi+kDFjVDlTgNoX8/xK2mGnrg3Eh5XAGDp95uqZifu0MGZ2MUEhjJTjw/7lUxdO
+nVXzEPf5FhuWHXx3Jmuy18pvDume+Pwww1r/R6+9w8k6qJaC/fgKlr3pVxqf83Bk
+owsdeGaj24mdhz/rjv82vTdgimWr98O3tkiTUAX3J7T+Bc/SpFqLjB+rdki4yAyX
+kgxDQa9ZpMOz0y9+NJYP8r7VH24pk9YVABYI8MHyLVIYoVew7LE0uVDRp7a8tP0L
+ClHOzemPgj6Z+HA9rSfqkh98txPpzhGJL0/j/XirY8vIp6vfkL+V8cYtGJ87lQvs
+pe24ZDWbSjCv0L560mNu48VRLGk1HwhzcapzVIdb2lX+tJCPBnakK23YXFZXqFXl
+LNwE3dGAPprxyJyqWj4Brgkz8qXrFDJIOATnh5trV8Ogy3ytM1xDB4ohRtjvPlUV
+zPzim2UwVV4oJfECAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAgEArhog4yLb8jn8x7FKArlSkjeQInXj
+ydZjT6SIxbNvRzw+RaqL6b30CFdc4csAcyJocsy+m+wc1xUwoPKcI3JwG/WPX5ni
+Qlx9dok6wuwIOX4b82oE/IItP0RPTJXkP+z+4i/wIfRhVywo199yJx1M3W3g7gBZ
+CkJn3XswdIpjiEwc5AgtRjmnL5gNfDjA4wYcKydMBmqjb3JoXKlN8KEzLtUz3XN/
+h2g4uGMLped83AJY3CHi/ZwQ/4EboIvbD4c8n3DQChBIUof11DmqAaqt1+CW7rzy
+xWPDWnnw6+9qLjD5WLyIrya6lZp4t1QCzQVFko/PIZkcO8/vrNmrvwzvqF34z0yY
+4GorVmlsHcjZUtpPVMMBNhDZweBOKT2LKIhlbnyU8GNWJPXz7QvdTMhdCMubPM3C
+eoS0aACiUM/K3wCE2ljXB8HKxnEXAAOJo2FwtL1ZvYEv3cKNcNHfb0AA3XJN1yMr
+du/99dph6/emeVnSDRI7sTQc12D8J9Ck6Mq+LDPQ7HvjeC6mvp7D9yMOqWgd4kd1
+QwL0u+qp3sFMKfcStgP/4dqZKwyjzuFS3fLrVXMGxZ46bTXVhITkSRSal7/WgCBx
+NXzYnP+S+0/hEg/7j6Vx904cvFNS7+jhUTUs2oU3lj9m4VZ2yE8w0XqYn/Cw+dE9
+AbOwQGKDn+Kq04s=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent2-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent2-key.pem
+++ node-18.8.0/test/fixtures/keys/agent2-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXAIBAAKBgQCrYG2i9zCUhQ1UCLe6RvJC95k2DdWSTEHoiH4IvvRURffSOLkS
-lCoaPCTlRvq1Fh+4YfAFZI7z5bY/C12mUNu4g6grkwCn7CQNEILwfWXiLxx04D3s
-R3DedFToyI+tSKKtXW7aWIdinpgMtaLHzRSI2cfoQ4sV3lbLmzsJK3isSQIDAQAB
-AoGAbaqNPiXUjpX+C3Jwr+FKkhQDlRWRP8dQvc7qaApaK7uCdKIbvInXz2YBbj7X
-nB4GOmVhxPGukODEmN9bFFzV3NbQVcAO8iPI6ldkYCmpmwfeEA1UJnBfBC2IYVFr
-0TSuq7OdniiO+FLhZvSAJXN+5yDv66nZEnojMuu1oqpG/gECQQDdxzSeNbOY2ak8
-Db2ZYKSBWjxVbW+UC8mYA7jHVpcUxlNnamcJrhg6hEfJqax6c/vAefECFsngG7yy
-XVTYX6rpAkEAxdI83zQz4NqYkZmFlCElNDRF7S1Sa2eX1HzzJYlOEMx3P1iE/z4U
-HOQh9py2jlY1th3GdbChF1VWMIOqGCtaYQJAJwxS/GQyKgBw5qz4rA+zBz9vDg+F
-rMhih0xodViOo07EEppOaArqIyt1RFGGl8ziD6Kox5hhlP7tO25paOt3OQJBALSL
-6y60EF06ZWENwxKtJa19wAx1/vEz/SjcWXZ62JsQYg2Yltn2KJktxam04hEKsb7j
-cgxcBsqrAh0JLicc+kECQDmH1wTvulw3E59jqOaKEbYNQFi18zzFkIIoNRVqMhtt
-zTJx8NYT9NUS3YE4OcUX0dQtVO3W+NIVrniaY8i29UM=
+MIIJKQIBAAKCAgEAzFbysXv+9Krce+YF58ByJq4vrQ1/0t2jHVlcWmAVONDpG4nt
+F8y02xlkwmZGLIX0EoQAbSI7gvJT+s3GEc2WHdH9rTQu4vqGTRdec7bcBZryoW9m
+ZiZyc5xEsXEWcFzZNhnrRbUPeJ7jsR8qAdAkDzRVW4uCKWjdbetVDGbsEdO5kzja
+L2rmLeYq17x+obGuvSzuzV4lZgAujQgug7bapQjgRvVbwPi+kDFjVDlTgNoX8/xK
+2mGnrg3Eh5XAGDp95uqZifu0MGZ2MUEhjJTjw/7lUxdOnVXzEPf5FhuWHXx3Jmuy
+18pvDume+Pwww1r/R6+9w8k6qJaC/fgKlr3pVxqf83BkowsdeGaj24mdhz/rjv82
+vTdgimWr98O3tkiTUAX3J7T+Bc/SpFqLjB+rdki4yAyXkgxDQa9ZpMOz0y9+NJYP
+8r7VH24pk9YVABYI8MHyLVIYoVew7LE0uVDRp7a8tP0LClHOzemPgj6Z+HA9rSfq
+kh98txPpzhGJL0/j/XirY8vIp6vfkL+V8cYtGJ87lQvspe24ZDWbSjCv0L560mNu
+48VRLGk1HwhzcapzVIdb2lX+tJCPBnakK23YXFZXqFXlLNwE3dGAPprxyJyqWj4B
+rgkz8qXrFDJIOATnh5trV8Ogy3ytM1xDB4ohRtjvPlUVzPzim2UwVV4oJfECAwEA
+AQKCAgEAtZ9WydCHldrEbM/QUeQo0h6iu7My+S8My4pg3udmFYECoPqgJTZLTi8s
+8ulK2Eob3dU8N8RklwhjApqhwbM1Cb5bNvcZ8Z9xfdYWms1elRDbNgg5QPRNaFru
+Z7U3l5m/aynwh53eaXz4rrcfdEsXsff3Io5GnB3cKtYwWvykOVgyKxwuOlVO0F8z
+l1uwRnIS+YyVmS5kGVxyA+LQzTva7TONfSSacFPntWxgeD0X7oBrJ+8IbUC9dKRC
+Ga1N93sN6xB8djumNhrsVO0qL7UrGUtXc9pUgt4hVOTDnhXZAM5fbWPuq4T/34Zh
+SbeGdC1Ol94p0nrqy2w0dB5i8wdL/QvDhKJJYBYfhkKMztYSdaU3Z5LdBJmHARVL
+WRY1J/kzIjXtXibeJYoAwjLRLMVx0fgDCIvV6Y7tKml6gwRHlB3rSxfNImibMzSX
+kcMEvM6MokeVcLNyBchGtmHjZsWDHba0/js2GyZmDNM7Hzn9ZOC9GRRQg7V+1osX
+zuHMQg38dLMulRdFWLdmOo7m2VvvONdDDSP4MiyLj4MFrRSMpRHI03ZxPV7fv3ef
+ASM0xmMhkxqWK7kcPKz5AVI28vYd7kVFUaIH9JqerXTHlRsB7xDkfi6+j089n/Qi
+Tw4uWLez7HuiJ5xHxWRMTehOuidQ3f+GV42zq+LJJrre1CPmVM0CggEBAPEgZ5ag
+q8Bkw5vIwTKO5ntVAQ2qDNFQdMgCL8pMDOcWuFvU60pYMnExMdIf9cErAKJO3P4d
+UJAlwCQ8rFO9/Z+JOh3LECOmtgfayVALaIIyRQCFVsOd+LyP9tCtAH7L5OrncuGz
+nixFPVINco0ZxT+XVzQeB45M1LQA8IZgmoKZrWgXuJUhzW09uyn5OP9HKjJlWTtZ
+/rHrKlUuiUKJ+eBeC79nldwFqvgQZfRo7tgiF7SquGWtn836V9h6w00ZjWeFB7gV
+RV46jRp3vL2dETBuleGg3O01rv5NPttjpYVKOMZm1QpQgTD1pT9DlsquNxkIuHRr
+pvkWNpgxPonoYs8CggEBANjxpWTYzMrPFS2jDNygT8YORN/2pkDOQJ8vsVEZgUOE
+rDE7H44ZXexMnlrkpcvl999SWLhNhQiAzfGYYeCwIsEddoeLWH4kcNhlqPMGktsy
+74EjvjODTnjkK0Z8SKqIFAMunzEN+bM5WJRGEzRMupwrAqkn0PVML7tVI7rYeVVg
+alsbSBOpvWzCjJhujbRoJAWNJHO39bm0TrWwb+X/KH+pDlld9PlIWUNoy3BQaoTU
+LumwRjQDladLh6BcwkgynQMDoAspCmoDN+9rAyN/anHWt7huqP6mTDDCOuRHn0Pd
+XYe5UuKFkctG1Akssh6zow+LpKCs/UTVBdfAnO60Gz8CggEAJs4TPa9/1Iei30hQ
+p0VOHF0SvVM3mtv631FSxZi1nt8OLmE2uJE3FPZY0Tspfrvf7z8H27F2DKy2nl5R
+UWzLC0i3XKIYZ9RliSARs9sSJ5nVdv7+dp7Or/HvQ6gD42QG5DL5Tw9sDi3ab4EQ
+hVq3XsOJWi7XAnSk6o8YJlgXpZr7yDvBDecRu1nq7Bdar+KRDOk4kg+qqJFzD6Bc
+oKO53fzkoRoXiQA0gpWJAut8PLZH4GB1Ko/lnu2CA6nGEnthQV6sXli2ADMmEDNi
+4ZUMdiHl5xjPVTGFhxGBYydHNQs71doowO8xy2JNBMBo3RdqcJHxtXK5vW5wvM94
+BBOtCwKCAQEAjpvi1nUxwkv4pwRc6xcy+YXdvr/7r/QlRL9sN1ylvYWUT9SGvLlu
+JWyEWEmft3ZXx+Twz06eGeXZ+2vpQ3NllT7R8O4n4NwBli7VrdPva8qC6Ihfeqth
+RVTwIaiHeBp11VWrbeq67xbOZMHf0+xwa0JWwG2dDfLkGagxEZ1FvVrg7zzy14Io
+NDt7gqSB4ErpCwEqKiuOi2G8r/Doze86ZxjxhsudwtTNYOOdgBpMuz5kDFhE/eIx
+WUTsRf1T9XPYBOg4J731y5SMC4V0Dbyvc1/ra4DK9ILK6O51GiT+y/TTi+nGU4Ps
+QA0yS0mnWUlK8mggc0LkRoAmFBrYzo10rwKCAQA5Nzf35c5E3Y6Pnnwvdj7HeFzn
+/pCks+/8k5RY7+Q9EGp/4fUpUldtgKSZviTd0A/xey7JCwltiGfCimdTb3y1nXIr
+se+uZymmA6MCMfLsd5tWOxF2sIdJUeVWKnlHY+kJ79VcxFlKPvw1Gr35oBfBIJLK
+0zmnNaOXnP6EWLNw1yZ73/mb9D+eL11NMlGiSQSLHkBxOFcfhY2bAAV+s2tbEafb
+QSPIcfe6zwbYa6XymYxNPUPvbEPCTRvv5XFgUTK0LyOKrWUFkVxqOIdu/+KIdtl+
+Xf8sOw3IlduzfZNxt2mgaV69TzAXXMF+tOY7PUHu/13al3FPeyzhGA3WHjHg
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent3-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent3-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent3-cert.pem
@@ -1,16 +1,24 @@
 -----BEGIN CERTIFICATE-----
-MIICcDCCAdkCCQCR8AZjYGnynDANBgkqhkiG9w0BAQsFADB6MQswCQYDVQQGEwJV
+MIID9DCCA10CCQCR8AZjYGnynzANBgkqhkiG9w0BAQsFADB6MQswCQYDVQQGEwJV
 UzELMAkGA1UECAwCQ0ExCzAJBgNVBAcMAlNGMQ8wDQYDVQQKDAZKb3llbnQxEDAO
 BgNVBAsMB05vZGUuanMxDDAKBgNVBAMMA2NhMjEgMB4GCSqGSIb3DQEJARYRcnlA
-dGlueWNsb3Vkcy5vcmcwIBcNMTgxMTE2MTg0MjIxWhgPMjI5MjA4MzAxODQyMjFa
+dGlueWNsb3Vkcy5vcmcwIBcNMjIwOTAzMTEwOTExWhgPMjI5NjA2MTcxMTA5MTFa
 MH0xCzAJBgNVBAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNV
 BAoMBkpveWVudDEQMA4GA1UECwwHTm9kZS5qczEPMA0GA1UEAwwGYWdlbnQzMSAw
-HgYJKoZIhvcNAQkBFhFyeUB0aW55Y2xvdWRzLm9yZzCBnzANBgkqhkiG9w0BAQEF
-AAOBjQAwgYkCgYEAv+hwfSVuJfHDQOXmF2D/HsI2JZkspfrPQE/ZL1tII2cXRnus
-IqWZvLg9v7IVY0gSvx5gWMHxmqqaK75McVJvO1XEzLYpa9Ddnj06xqNWl6hwLHnP
-bclRi2n63Cs6zSM80r1iQ16ovZ0hyWPjXaBlWmb71QeeBp6ynxhB+yA0eZsCAwEA
-ATANBgkqhkiG9w0BAQsFAAOBgQA/C2xJIYA3Vo8pr1cfmzN+os9uvbMQEAegg6W+
-6/t82tLGnrCEglMTFHFp7MJCNiKCY16Ixi5WCZoUrGCfbh+Obtd4bP/wAlR8AS67
-lYZDvjADGU4e7Aqu0o0AHeb4MiRUQbkD0EUioyD8091Qhzlrx43UtdojPvakwAXM
-N/LFEw==
+HgYJKoZIhvcNAQkBFhFyeUB0aW55Y2xvdWRzLm9yZzCCAiIwDQYJKoZIhvcNAQEB
+BQADggIPADCCAgoCggIBAJ+hODcaGManC6vB09X8HsPFpDu/n9fWhoj23dFJb02c
+gvjwSqHE+F5is/WSxZ8unTYaDQycW7xIdn43LbFpmUsQIWnDWypjLHvnX3lmG6HU
+vAkfBVt4UMvKq8bycdTQT0jJcvhcB1Hifwkm6soST/ssUa52jryKWMkYm/VElIm3
+99oV6fcXFrKNrhtx2OavI4ftVVc8OYJBFzPa6dFvmIiE1v2xyEmE+N8dzgl1Jar8
+5l/4vdAWCn7MmKkgBBVxaNQKv6dY/Cf6Bc0krCx2B0enTQc7/Fz8Nzqt/Hw6u3GF
+v+Bw9X1Bo8+56b3gzJD/xXLwWwQdUJPwNGjeAUf5jImccvC6joGuc43Rh+Gh8HnP
+f7tNg9r9P+vhaF9vBXd/cRyrmQhanSk+zkQdL3fCsPn/wQGyxlB8ZhPnmzJTQEXf
+h+bHUSH8JDvnPdu4GDN6PzNkCsRtLDW+mk1WvO/9fdxd5RVXZCDicVFh9hFwHEeP
++qWwlOA2pJPAdEeSmpwIIS3DE5k1UBOSllqMYzUKnH/j8hSaGT5+CvOB1G4r5p/g
+borE4br+wdoOG91tvYxCqbYn/DOPun20kXkg8RQM4cT98QZMU5l7ZL2DveXQ0H3H
+tOPkxB+3bpOTBVN1Yse/4uJ5xIF+8QTsBgbtA9VsqZJyp0XmWqebYxEy4DxyiiMv
+AgMBAAEwDQYJKoZIhvcNAQELBQADgYEAZvaTyenSVZgGG+o8UvRwlKycjAZnm5n/
+SPwS+cqkJVYwpnCsPUho9wY3AMI6+DhDh7KaAG2TnkPBIjpLIMtGruYX8sRZUKOp
+KBuglQw9C22sQUDpA2HYstIrjCao4nUQYMCTq1M7lBmkmAAIbLp4IF9UPFyl+vhr
+Nb6T5PDR73s=
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent3-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent3-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent3-csr.pem
@@ -1,13 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB4jCCAUsCAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
+MIIE5zCCAs8CAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQD
-DAZhZ2VudDMxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0G
-CSqGSIb3DQEBAQUAA4GNADCBiQKBgQC/6HB9JW4l8cNA5eYXYP8ewjYlmSyl+s9A
-T9kvW0gjZxdGe6wipZm8uD2/shVjSBK/HmBYwfGaqporvkxxUm87VcTMtilr0N2e
-PTrGo1aXqHAsec9tyVGLafrcKzrNIzzSvWJDXqi9nSHJY+NdoGVaZvvVB54GnrKf
-GEH7IDR5mwIDAQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3
-b3JkMA0GCSqGSIb3DQEBCwUAA4GBAFHJUONDqOhZpGN8ZCFkWkGyD4iDPGdJyR1f
-lh1N2vSf9vx663ni6lG9XQrQZXyPH8n7vvyyX1bJE5X6dAKuiD4GYlcGUUCnsvcA
-r+JzSBrbtwD57bPnn21YSUl2QEoG2b+/6uPKWxKr8e1sreMxHOLwsPgSavnQ84Bc
-GvSLlIcR
+DAZhZ2VudDMxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAn6E4NxoYxqcLq8HT1fwew8WkO7+f
+19aGiPbd0UlvTZyC+PBKocT4XmKz9ZLFny6dNhoNDJxbvEh2fjctsWmZSxAhacNb
+KmMse+dfeWYbodS8CR8FW3hQy8qrxvJx1NBPSMly+FwHUeJ/CSbqyhJP+yxRrnaO
+vIpYyRib9USUibf32hXp9xcWso2uG3HY5q8jh+1VVzw5gkEXM9rp0W+YiITW/bHI
+SYT43x3OCXUlqvzmX/i90BYKfsyYqSAEFXFo1Aq/p1j8J/oFzSSsLHYHR6dNBzv8
+XPw3Oq38fDq7cYW/4HD1fUGjz7npveDMkP/FcvBbBB1Qk/A0aN4BR/mMiZxy8LqO
+ga5zjdGH4aHwec9/u02D2v0/6+FoX28Fd39xHKuZCFqdKT7ORB0vd8Kw+f/BAbLG
+UHxmE+ebMlNARd+H5sdRIfwkO+c927gYM3o/M2QKxG0sNb6aTVa87/193F3lFVdk
+IOJxUWH2EXAcR4/6pbCU4Dakk8B0R5KanAghLcMTmTVQE5KWWoxjNQqcf+PyFJoZ
+Pn4K84HUbivmn+BuisThuv7B2g4b3W29jEKptif8M4+6fbSReSDxFAzhxP3xBkxT
+mXtkvYO95dDQfce04+TEH7duk5MFU3Vix7/i4nnEgX7xBOwGBu0D1WypknKnReZa
+p5tjETLgPHKKIy8CAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAgEAOK0lgzISKQfsE4pun8iVjdrbF92R
+evMV+LpRhoFFCXDF4ksEDUTMTOpaqrQ7H4LPRnsjRY2B6JRRiyZph72BnE+rSfy+
+whJKTkANrsReBka+Smd9QTq4sX30cyhHXykR2eVLO2rpbrc7M09KYyrOhKv6akfn
+ZNlmz4UVu8cnlLPRAV7QGvPM0oZ7fLixa7H2WAtUeAf6XWgbCfM7RgqAFI+LNVVi
+TNXKeuKxWrUZJ3kXoEZyycqDBsViT0LKeXZjs4j9JRK1VSz3M2iAjMlhJeYWHwCB
+rcLCn9VtxvGkUWtrnZYO9MJTtdviluJPho+6sD9Fs2rtUIVYeBaOAg/wau+jMB/d
+jpuZlwAvE7HuFUIGU+ydASL4cnH11x87z2AsKGnTjfMRuOSdLbA9p9rfXLP0cKmW
+GUNXdGrWoxI9NceEZ74CapUqhC+WNMmtPKUmOl0O8aeUt2UZfMTstKrWWvK8HGbN
+FfsZCtI/Gyaou6HeYmmmLptgYeme0aLDsUScBoBVD1bHGhaaQM87RyGtgIgrBktc
+d/ilrkhIklTj3+o8mH7CCfaHvUEbe/0od2lMDkIiSZdSqmcnu+VrD6F5lhmriGX6
+oQBI5i4By5J4fiAavDSpqjh1bzM+/WTKxH/g81t7VvnF/M+FviqovnrWtgxo9ziN
+yD47tTUxh+FyJ9k=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent3-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent3-key.pem
+++ node-18.8.0/test/fixtures/keys/agent3-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXAIBAAKBgQC/6HB9JW4l8cNA5eYXYP8ewjYlmSyl+s9AT9kvW0gjZxdGe6wi
-pZm8uD2/shVjSBK/HmBYwfGaqporvkxxUm87VcTMtilr0N2ePTrGo1aXqHAsec9t
-yVGLafrcKzrNIzzSvWJDXqi9nSHJY+NdoGVaZvvVB54GnrKfGEH7IDR5mwIDAQAB
-AoGAa94V5HH2kMNskXznsPpnS/2z+7w2OXFZrvdyx0iSqruWfJqlLbBRUp9orehG
-V1C6oMxNMXaJ+/qqv62uQAAq3oCPHRrN5a1fLzYKk/ixUbk0F7saNvOsmWqbSIzv
-OEtsHBt3zJxEkgLFzuaFnfoBoFL7lvJYol+4QPVvxYj2exkCQQDqVRGFbhRmKBZ+
-ienF9JpUOruKEsW4lmSKP2fUAL4rH40cJEFD0OI80/WdN/34USPOEqSsFTZITvFH
-+y+45PD/AkEA0acinvAAb0FLlMlhcG0LGzIwcYUWcEpPxwkioGgwGaEPLR276gHv
-NvgtL7xgLi4QMKB0n48zz4W4Usww6QmbZQJBAOGIZoipXfDEfIHlcp4XwcF3lbBa
-SPpTpQh55hBhdqZCg6mmKzp9/IDW7/oVPdaVIYTg5KTK9ae6cvb4hwHJNzkCQE3p
-YclVAaRWzKK/b/Ga5Gy36x7UybDzPNCHyZF5Bp8PppcqnKHrFB4GfqxlwgyHW8bm
-alC9pBBz7jr+3RJNWq0CQFHxhrjdhQFhYJoW8b+pqE1ryNSSSf/1yBEug6Xsqsjn
-MSuyCTLXRpoS/LAdL95ENCX3ULsu/5nlpKZmy99yY2M=
+MIIJKgIBAAKCAgEAn6E4NxoYxqcLq8HT1fwew8WkO7+f19aGiPbd0UlvTZyC+PBK
+ocT4XmKz9ZLFny6dNhoNDJxbvEh2fjctsWmZSxAhacNbKmMse+dfeWYbodS8CR8F
+W3hQy8qrxvJx1NBPSMly+FwHUeJ/CSbqyhJP+yxRrnaOvIpYyRib9USUibf32hXp
+9xcWso2uG3HY5q8jh+1VVzw5gkEXM9rp0W+YiITW/bHISYT43x3OCXUlqvzmX/i9
+0BYKfsyYqSAEFXFo1Aq/p1j8J/oFzSSsLHYHR6dNBzv8XPw3Oq38fDq7cYW/4HD1
+fUGjz7npveDMkP/FcvBbBB1Qk/A0aN4BR/mMiZxy8LqOga5zjdGH4aHwec9/u02D
+2v0/6+FoX28Fd39xHKuZCFqdKT7ORB0vd8Kw+f/BAbLGUHxmE+ebMlNARd+H5sdR
+IfwkO+c927gYM3o/M2QKxG0sNb6aTVa87/193F3lFVdkIOJxUWH2EXAcR4/6pbCU
+4Dakk8B0R5KanAghLcMTmTVQE5KWWoxjNQqcf+PyFJoZPn4K84HUbivmn+BuisTh
+uv7B2g4b3W29jEKptif8M4+6fbSReSDxFAzhxP3xBkxTmXtkvYO95dDQfce04+TE
+H7duk5MFU3Vix7/i4nnEgX7xBOwGBu0D1WypknKnReZap5tjETLgPHKKIy8CAwEA
+AQKCAgAIdZG2EdL4TAfpTTc9osO6mjGu3OPKm1q7zWFJ4vtcAIxMRuedlQOYdOCW
+hfWlDuJT6KOCZJLT8n8dBy2Ltk3iVXN/2FNvdpv34PYFR+z08YIZdaRiCYk2k4Hw
+5A42OefJsqs+EBpvM3KJ+J7+oGPOqrWVakD8KtSzq3MwVzcL6UzgKMMiTf3s7mco
+myrkZqmaXXBlVBK0HWDuZY54g9JwgaGnBhHYG4zYvgqaI7QgUJCgy1y7tz9NIuiA
+Jesxe7YTbpJjeuq575z5FxVsscCHAo2pcXtZLiB7Ipc7uh0O0giuCy5nOeBFdNO3
+NT+B+5vqEGSyXh9UB6mvWPx2ZPoWh19x3wnMmm4Y/a1zTjR3/r7zowFRzT7XeH9N
+DVQsAYRhYl6gmzuaCeFFVSruS/Ziu9KuQHlqRUs/bijoKasTXXbA2TQISqyVKl8d
+bu5BhzmLLCZmPo+XeLoKuildMoINUOitqWMcsL3ShTVrXEy0fv5LKmOmsCJ2qnU1
+/95f1bH6IrBTnUKCGbPd6SRk1JRxp8dmO5ufgKgF2oGW7QR6o8EXiCUjWVmzDItx
+eDTIMZQkJc0PhW2wNOVMqnZWNWjsNOlbcSzL23nN7hyXCmQFTbY290PS1lOzPGyM
+MCcQ/CRUUu4n38KqfDBht6Lm2rG03rqva5aay+V8nEqkV6UNcQKCAQEAzQE6xipj
+af4zcMT84qVDh8eZkjaJhmxCoumWjX9saCj/3EklJzwtw/XYSMQFcPMbrPvV4q/p
+XNScpAS8oAmvtjqqJ5VrN9Pvzgs+p8V9+DF5ucQHVHVB8ZiOHz9Fe1Nx4IdjJY3A
+S0UtwE2UHy5ro7ZFMvGgeiGJKHjxXAX2jjU3Fj4jT0ByjhJ1uMwSLzCJrujVZzOR
+Bi5cTkycP5FshRmQa/Yv0NRugTZeeyTBOZ3GsYWAfwJETMNWS/qKEetAw1uHlgo2
+takAEZGGTFsJIBS+p/uwhSeBjeI6epVTB6Y1MeaV/HBdHUP84ZOdWPApcRpAOEZN
+5So/lOIOq2VsFwKCAQEAx1Z+RFGRwaR/UvHZWOzYuXXDYlXiC2G5XKj7GxNltI20
+FY7pn7gpvDRlwMsEIOT1O3JjO4J7kBZjVPVzOaU5nRf+fQYgRQXvJtBL+k5mYVIG
+yjmWxP2QyGYbyIxhw8AjEO/S6vGSNYplmdlkWUDybVn2cX0iaj0amEkkc7vPHj0P
+BkkXblg1T0LVyelmc9Y8gvH0AzdqXsLT5osIDm5vahgSNl+tMrcQuEa4ZQ1gDapv
+bh5xbRCOuGiFLjo6+vQJ4UcdFC1yYRGCoWJRll2+PRMC//yKdWOhAt40t6H0vb4B
+lmhwMlafMbBIG8G1Mfp0sa4ubBsagQ4aPUPKGU14qQKCAQEAk03l4IyGwJuSdN5m
+46hifMcUdzZO3VMfEpaLwoKj77g+GRhJ5QZd0Cfv0GFhk4ZsBYEzb9587WIsZ8Kr
+ksdDBcrjhuvbjdLwxfnYfI7w9A0A7VWI+swCpUm8HYHGOFZ+RhqiSFKxsKQ1uq9D
+1p/xYkJPOMrcR8WfG4DD1R7Hyek8+vONRZ4GckWbZzzjeBsFY7QT20L3Wxym4PA1
+eYvSEVWMK7dLJl/uQ1RA0XVYjkDri5Vk/l4vm4TkllWhLjz/XSCArXQLWb7kuLr7
+aBQl23JsiKHWdecsNTowUPLJJX7tTDzm2uwWTDgVFjWOY2C2NnnpWUZTJG6WHzHL
+3qZS4wKCAQEAwxzyMsT0umndBr8bU2NhLwcJ2/Lccu+QlBYC05nE71nUb1WXb3K/
+xywyfOc6f0gckbeBUfqCH6MKT1qBEmebYRzfSFj6tRT95Sv1cmfS5md3pfGAlgjr
+8nufrklaWFR0/lsARF94EVgJ4vFau9JKeKaaBrPJP9vXcEAHohCuxpZVYn2gpa1J
+YO+2exAt7CxCxpk1kwBm6aIh74Meu6F1BI/TI826pe1yPC1cx+yznCy7LTo3I94V
+FIOZ+ImeuYytLwiAegOR9VlCV9z4pwnS78zwHsh+bjC6pSgCycuI10BoJWTMjTkk
+UWqO6jxMs0LHfjNT1RUieDfpjxYPd3zRIQKCAQEAiwhlC5Pr5tKUu2/qStQgnTdw
+qXZZNb/ElD3qy2jcHWbNXA4ZJ1IDGJh7SXfdrHxNUCmaZ8pQ0sppgUWlVnizvO7+
+Zo2vQK/zNsdAa2i5dph5Py9KA9DqGNnOJjXx1sj/7Spxf4BHZjNRJeG+x/+NDZqq
+on7Px+5tJxJtruh6AXQoWIpXqs1ml6tiAh/UQNiH6t7A5fxYg8Bhfp0BJXT1XavR
+C5SqeNOiqrQOuscqqsM2hO2Kh/NmC3In9PNfCUeV9gCcZiJvJhdtJAzlaSdvjeg6
+iIWZyIpimqwJ+o8cr4Em5s01yGQHyEI4WJesK3LIlveG2P3dwQ4CaWaP7Y1TTQ==
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent4-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent4-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent4-cert.pem
@@ -1,16 +1,24 @@
 -----BEGIN CERTIFICATE-----
-MIICjjCCAfegAwIBAgIJAJHwBmNgafKaMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
+MIIEEjCCA3ugAwIBAgIJAJHwBmNgafKeMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
 BAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoMBkpveWVu
 dDEQMA4GA1UECwwHTm9kZS5qczEMMAoGA1UEAwwDY2EyMSAwHgYJKoZIhvcNAQkB
-FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0xODExMTYxODQyMjBaGA8yMjkyMDgzMDE4
-NDIyMFowfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQHDAJTRjEP
+FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0yMjA5MDMxMTA5MDhaGA8yMjk2MDYxNzEx
+MDkwOFowfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQHDAJTRjEP
 MA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQDDAZhZ2Vu
-dDQxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0GCSqGSIb3
-DQEBAQUAA4GNADCBiQKBgQCvcVH69FzdPGCUXgwowuBz4lLAV+COzishbuyNGF5l
-J6mw6eY8gizLmpxh0r6d/REnlzKRy3Uy9FdZEQZKKfeK63MxLU6BYaHX0fnqz2y1
-oCaA2eW4yeGOLaSBcjEKHIs964Ik9VKEDnioYtoObbFihRbcS6QLNajQR9ij+7hl
-pQIDAQABoxcwFTATBgNVHSUEDDAKBggrBgEFBQcDAjANBgkqhkiG9w0BAQsFAAOB
-gQAykzKtLG++xAfLoq2AZ3OKlEYGC4RuOCy9J75RYWNRuy18UnYl0HGSepeqQ/4c
-0r+dy/SLUVKxC7e87hs7gP8UX+f9UaVM7dvqvZbZMVH+A6w2nIAcO3zwtSJlfQ8H
-NJAdQl1lZ6qc97APtBlfeTMTdi/hTghqZLah21/hIE5lFw==
+dDQxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjANBgkqhkiG
+9w0BAQEFAAOCAg8AMIICCgKCAgEApNuN4v7Hazql+8SlRqZ5Cxo+ewOIJ6A+vj7s
+dazpnp53HNuqgx2BUG7XMOTlwkd2PrAu8Ff77b3epDzsxwxU8BOCp441ue+U7T8/
+zAGBTtVngtRsFVyNTZg+AhxZWv1r1o0nQZ+uq9ViMj0wJGpoVrhWm3TVq7DCz/Qz
+Tx9I7vmTN4iCZkfSpd4WS9HI7hkI2WHmTRi562jle5meoHGp7Aq5kxFyX0Ajj8YL
+Z8ZshmTuSWfWAasdQZfUkF9WCpIrjo+wwdIsfNsFsEDGFA6Ji2tc0uqfZyk5zC2o
+IJELtIC6jo1Cza7+pMkIATV6ztrnliuAOkWKLI2A/ZloadVqPMouj1WkePAnmpj7
+OyHwZWs2bnGvVa5CWQvcXiFOzLikKLUOV6fZunq9kEaJFqEnAQD3xTCBZnoa6/hX
+P3pKQHt1rzZBvsmXv2zj+B7tTydbN55Q2fwrIbP1ZV3OTPNdyh1ebEEKNX0rllnO
+xwa048Cl+fPzCXckFYy+RIDZhUVFFPy03JxUVOlpo/NMur+G1gq5UgNYgzvyfXLS
+x1B7V+fGKp/pDbiwc6SK8sX5a9WKSlfZg0HtgIkk9JjcA10uhP29nAQwGqxNkniv
+sdDPs6ZYy/xTSIO8F0bnfCcCx6YMb3gg3Y4UZjWKSx/fOSXCPx54H8aDcYXyN2+p
+qIgKITUCAwEAAaMXMBUwEwYDVR0lBAwwCgYIKwYBBQUHAwIwDQYJKoZIhvcNAQEL
+BQADgYEAHtF/BAWE6Lw4jN2RykAmyzn5ra9GMYI9cqH54BiJb773Z7IIu3XaWL2c
+HFAW/Zd+TOxKuU+BfKOQOmOgf5bwO5m6WBvLqSdyAgK7y5z8T8zzoz8OzVvg/V8g
+83OFWtQdjpIdoSD/ayrs93wqN0N/x65+j5NJFJYs4r++v7zGbe0=
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent4-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent4-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent4-csr.pem
@@ -1,13 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB4jCCAUsCAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
+MIIE5zCCAs8CAQAwfTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGSm95ZW50MRAwDgYDVQQLDAdOb2RlLmpzMQ8wDQYDVQQD
-DAZhZ2VudDQxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIGfMA0G
-CSqGSIb3DQEBAQUAA4GNADCBiQKBgQCvcVH69FzdPGCUXgwowuBz4lLAV+COzish
-buyNGF5lJ6mw6eY8gizLmpxh0r6d/REnlzKRy3Uy9FdZEQZKKfeK63MxLU6BYaHX
-0fnqz2y1oCaA2eW4yeGOLaSBcjEKHIs964Ik9VKEDnioYtoObbFihRbcS6QLNajQ
-R9ij+7hlpQIDAQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3
-b3JkMA0GCSqGSIb3DQEBCwUAA4GBAJ4tZ0hFXYqGQ0BDpcI6QIjufzoFHXMHBmE0
-wHU1f8jVM2v9Df5eInArMvAVya4gXtuZnMpRZKNrcbnwPUK9spwIzHxPyw7qjeCP
-SG+TusJoFFIGgpZBo6zVdtpRCRbTxNfKteK+y34g+sYZolt88AmlzY8H2QYeQabI
-1SBuLdBH
+DAZhZ2VudDQxIDAeBgkqhkiG9w0BCQEWEXJ5QHRpbnljbG91ZHMub3JnMIICIjAN
+BgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEApNuN4v7Hazql+8SlRqZ5Cxo+ewOI
+J6A+vj7sdazpnp53HNuqgx2BUG7XMOTlwkd2PrAu8Ff77b3epDzsxwxU8BOCp441
+ue+U7T8/zAGBTtVngtRsFVyNTZg+AhxZWv1r1o0nQZ+uq9ViMj0wJGpoVrhWm3TV
+q7DCz/QzTx9I7vmTN4iCZkfSpd4WS9HI7hkI2WHmTRi562jle5meoHGp7Aq5kxFy
+X0Ajj8YLZ8ZshmTuSWfWAasdQZfUkF9WCpIrjo+wwdIsfNsFsEDGFA6Ji2tc0uqf
+Zyk5zC2oIJELtIC6jo1Cza7+pMkIATV6ztrnliuAOkWKLI2A/ZloadVqPMouj1Wk
+ePAnmpj7OyHwZWs2bnGvVa5CWQvcXiFOzLikKLUOV6fZunq9kEaJFqEnAQD3xTCB
+Znoa6/hXP3pKQHt1rzZBvsmXv2zj+B7tTydbN55Q2fwrIbP1ZV3OTPNdyh1ebEEK
+NX0rllnOxwa048Cl+fPzCXckFYy+RIDZhUVFFPy03JxUVOlpo/NMur+G1gq5UgNY
+gzvyfXLSx1B7V+fGKp/pDbiwc6SK8sX5a9WKSlfZg0HtgIkk9JjcA10uhP29nAQw
+GqxNknivsdDPs6ZYy/xTSIO8F0bnfCcCx6YMb3gg3Y4UZjWKSx/fOSXCPx54H8aD
+cYXyN2+pqIgKITUCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAgEAQpI1UGlgAzV6rSt8EmKojC/xE9O2
+Kj8OWKVCXUisJd7q8HmXLgF6ffGg8EElpWlGXqCDEQkSnD2GPEIMUGWlVoES42pL
+ox0IolKivdDU9eMrOASiOcgeviRZufM6i13cztfXsl8kdA5CeIy3Z7TReTWU7i+3
+stu4lS7kPN60NsioLbVZMZgkAsimWHNquzoxdd1RNf2t2pbx/QBY3vnXEYfBlm6n
+74V3lfkcbjbe4vnNLVDOEz8hLj5wle4/yBmeU6YAWr9/30jjyJvbqqZ8C4xV7pA7
+wFiOjZWeDPb3+iql8WjdMql9y9utf7RS9kao9f+G5X6rIvRE/LNPaxAOqPLxpIvd
+QywnO6HrooOkpkE+KaSmw5BPkrMcVsG3YnRBXbKuONEOUC6Sxa0JGBhzv6doMO++
+cBWWsmQkiXa1f6DVKJpAohasbMCTx0I9VVBFyv2FtZLKcvN44ucp02teAyU5rdUO
+VJiYtTk4rORbyOEqUf4+lgx7Zq3DoTakmmtwf7s3F/EYhlEqTLFSW6AGjGSk5HP+
+V1+EqfII8QNsBsfQ2L0vcU48B1v9fbCTZMMgvicpvEJJYToV1CvbuZ10KVQiBe53
+Rh++BXcScvn9dV7CHfNdA4WyAouMPcLX1eiC3yizX6rX+KMNTdULO3vS1zUVOV3H
+Tq95hW3fkeml220=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent4-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent4-key.pem
+++ node-18.8.0/test/fixtures/keys/agent4-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXQIBAAKBgQCvcVH69FzdPGCUXgwowuBz4lLAV+COzishbuyNGF5lJ6mw6eY8
-gizLmpxh0r6d/REnlzKRy3Uy9FdZEQZKKfeK63MxLU6BYaHX0fnqz2y1oCaA2eW4
-yeGOLaSBcjEKHIs964Ik9VKEDnioYtoObbFihRbcS6QLNajQR9ij+7hlpQIDAQAB
-AoGASgLVIbf6gP4ahByUppFnXJuqayfnPHcu7MC9m9we3i94P4C8tuP3f8Dunbno
-3f9HQFthYu3guCkpvBIZhCnmGgrge/Rm5IYN9Jktc8gTSQ0NlJbr3hjgHEkJXhca
-6zE/sFEgZVWF/yCIunyU3umbWBJE9R+suk+mpe1wZ9T+AXUCQQDU208b7AvW9Yoo
-8SomHE5B1FJOwnyEWHBJ9W0myGXOrJbf6zNJ4eOLQsn1UWSsI9tzgeh49f6smc7B
-bhWhHqoLAkEA0wCs6zyKs0pbzGfGYQFafDbjSUBbT+nn4tXR+3O7Z8K6x3gt7DBx
-VtlbJtfBBWCCrIgYsrU3TUwtweDV6umtDwJBAJN7tU+SeQ2jKex+VQb8+9gu5iy+
-IwqMQJluHQgPOENAYHWcAPiDNGdMiqSYldmUKrzY2RvezmwHUjPCM+hkV8sCQQCF
-MQL2RrQi8sg5ojQmXa1ZhWg5gAdjzXnTxTcUa/ybRd+TNDiAxB93PCL+xOiR1VcH
-Q62beSqcf37OyHcgHztfAkAEHZwjVSRXFELBQDPzespHXVC3rwpQlbd1tqOybmPd
-tpqmlWjvFxdvEDQZnemPUCtXNhMOaXcSOeeSYcUkIVX6
+MIIJKAIBAAKCAgEApNuN4v7Hazql+8SlRqZ5Cxo+ewOIJ6A+vj7sdazpnp53HNuq
+gx2BUG7XMOTlwkd2PrAu8Ff77b3epDzsxwxU8BOCp441ue+U7T8/zAGBTtVngtRs
+FVyNTZg+AhxZWv1r1o0nQZ+uq9ViMj0wJGpoVrhWm3TVq7DCz/QzTx9I7vmTN4iC
+ZkfSpd4WS9HI7hkI2WHmTRi562jle5meoHGp7Aq5kxFyX0Ajj8YLZ8ZshmTuSWfW
+AasdQZfUkF9WCpIrjo+wwdIsfNsFsEDGFA6Ji2tc0uqfZyk5zC2oIJELtIC6jo1C
+za7+pMkIATV6ztrnliuAOkWKLI2A/ZloadVqPMouj1WkePAnmpj7OyHwZWs2bnGv
+Va5CWQvcXiFOzLikKLUOV6fZunq9kEaJFqEnAQD3xTCBZnoa6/hXP3pKQHt1rzZB
+vsmXv2zj+B7tTydbN55Q2fwrIbP1ZV3OTPNdyh1ebEEKNX0rllnOxwa048Cl+fPz
+CXckFYy+RIDZhUVFFPy03JxUVOlpo/NMur+G1gq5UgNYgzvyfXLSx1B7V+fGKp/p
+Dbiwc6SK8sX5a9WKSlfZg0HtgIkk9JjcA10uhP29nAQwGqxNknivsdDPs6ZYy/xT
+SIO8F0bnfCcCx6YMb3gg3Y4UZjWKSx/fOSXCPx54H8aDcYXyN2+pqIgKITUCAwEA
+AQKCAgA+YQ05HSqb+l8f+vKG3yrh9pY5ucZlKA9VsT2L8Vecu11awAuu/RUXODmY
+uKPvdVbatZOtQb2D4dpCzwLQ9muVQ+ymNftCVdNGSi57pP4mtD0slfAnzRNyju63
+ob513XpWfqaiACf5wQ4c8GYtAU5lse6jd/v5sTKluUNI5w+TzT/VPYQJ65dbR8Os
+AL9EyJg+mPqppXeECfb7Y89P6nJfbFd3vJxISl6qYkeihsQ9vickA3RIwVSaoza9
+QjWuXky47sn1LO/EdqOaY4gPwshsxr0i2cBjKohCJ+x8FiYVsLfmhLEIXvTfqWmr
++25yowYKHn+wjOuRnCF40/lcQnKl3haI6dHP9WSNI3Gay3s93GR1c5Q2d1Ro9ZUW
+vILL9BEaQ6Hu4eu9mg3QebacJqW5KvtvllG7lFETqRq83q0Kx1OoJQXPZ+9mhuOJ
+FXqnLyfabNiSQYKIbPAQ8a+rOx3FXMeUvziQ2MC7ex/fC/C/bhOOig3WCjXMRJYv
+VJ+PM5FTZKIK3YgHFyo5gNFYexKxzZBiV6ah62Gpx1PaGbEUiRBtOQLF5lPX5cyP
+NEswluo1FSP9s2MB5cdLVzxDv6cAvsEu60pZTEICaSJLbX/3jM70Et0L5gWlt8Lf
+YIFYToBwmRHBU3hpoCFivaoLstLDrM2JcW79UxO4LceB2ipWaQKCAQEA20UkbOBW
+hZKbrCHnyJaZMXDP/LsqQMxgT9tDF2dKHusFdkVY1J0SD89+sKQdOQo7J6oEYP0o
+d5gzT72ZLI9+vrHH3ixVt5MHSkbas1LbWnIV++F1GKg48x+btOCg8xDv1jTOOvjk
+OYWongoZnoUChjvLC+b0cumDLfwuOaw211rzIau/3BYnkqAcehkvh/H5SdI9k8NZ
+CV0AOKkQof8ydajQobOAx0woPxLMb9T8Tdzt/ICIpy9ImyZwPGKrrckjKplvgT0B
+a4f0K7SN4NAKwPA+JsOTX1h0yngZphZrgOszFqfjhAgD5Ad8Mmh6Xaq8MKqsl0a2
+Vl/aYIio0B5/swKCAQEAwHkRYYWePlnrdI2HU5cFuxmxyCn5jLDACc0ShAO1fGoZ
+JDCjHQr8jj33aQSGJKPOFGOCULCs3LMTkJAsSXTfdwlS4unSD+d1PIul4qfYZMx1
+D6WEyYeOLlyHt2qIz5WA/tO18fGvkLojtKwlrTttiRbzE3FlJXjL3hpk9GpRzcW4
+01xdkNQboeMFaIfU1uFZik995kVuZNe2U3rR+zzXO+1tYtzNFM3BlR4T87b/lPHj
+OodXz4eYKApweQ2jdbi/+g+yEineXKxHHjVrqqxBoiLHgLD7ftj+dAFOB0PFLVGk
+LWsTyzkIJepCtEOWx5y4OQw2uHjjYm9Kz/6AebWndwKCAQBHApVRTTW3/Eq97B0M
+MPlbsZUp29NLOOTX98I6ihDPAn26V6r08AkxZZk5y2xLVKzjo4PXhSTd1kSHxncc
+nzwEtb5sjR8VXGA4Bco92M+A8wF7mQA7A/+tfMcIRNba4NlLHb7MCl6e4lJBHDQ/
+ygOPRT30lM8Fh+obslemBcec5q9mcCCqPeKC8adHQ3DJLev1pEic2d6z0ZKIxvfD
+/slwoUMvmtpRsLA3/PaIIQm0jZ/k7htTSVTSKhie9bVzingBC7+4svEKyfShUs+/
+xqA065i1JRFNB3CLQWnNB+W40nKxG53q3E2mQ2ZNTNiUDP71I9DRDO1v+4rVSGyj
+Ob1RAoIBAQCpKb8c91Mdeep3hexqi1F+IBcg1cN+49nkw1iCrr/TmWx/Yj/6Rq2y
+VsQciN1dVprJUKw4Pw+u6bVIWW6TISVGeetfhCIb2FIx5Irnqp2F+izok5EhjwJn
+vOuLSRb1W3EEhw5ilwi3KuI26RY0KwcadhEuBXvgc0q66nfQWaGMOTs4iPWzVLj0
+j5MxTDDEq0sHRNZijKN5/h1BQbFMHy0MfqClQ2r73dtGNClVCoGXZwFPb6S9wW75
+wjMYMeR4kugn36OiUyocSU44PrtiRgNpfSfa87IWyqDZVExAFlA2hIrHYFUFSBrh
+OANoa/8ZcaR4f0OY3qb63NXj+MryLbYBAoIBACkvjT8l0oHRlrBsiXq76I1iQjGn
+01pKAj56MUqkDyY9Gi6sljmzFZmYNclYLDm6Z0mC+zJfXQfcve1kTNCgBoan3Cvx
+2kHsCxsMhV3TWRYOcNlJ9MFV411j3irIUnjPQ+E53XxgJWmCE4oU5WoK0POvfjiq
+ubME68emiB8/u+Zkoa/qLObRpiAZgDIqeQcVmKS9u66u48xA5XDQE3ZvCxKDcVPE
+s2HnpzXP4cLomTIFKJ3UFvXe31rhdDon4SjzHu2ohbbhK/NqaTUHasm2PYReqqOU
+zLqWrJ+o+GgnjUfj02AOjRlwhHwEz2GD5SyARClxTCXdYxoS3HYfImbL35g=
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent5-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent5-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent5-cert.pem
@@ -1,16 +1,24 @@
 -----BEGIN CERTIFICATE-----
-MIIChTCCAe6gAwIBAgIJAJHwBmNgafKdMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
+MIIECTCCA3KgAwIBAgIJAJHwBmNgafKgMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
 BAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoMBkpveWVu
 dDEQMA4GA1UECwwHTm9kZS5qczEMMAoGA1UEAwwDY2EyMSAwHgYJKoZIhvcNAQkB
-FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0xODExMTYxODQyMjFaGA8yMjkyMDgzMDE4
-NDIyMVowdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREwDwYDVQQK
+FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0yMjA5MDMxMTA5MTFaGA8yMjk2MDYxNzEx
+MDkxMVowdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREwDwYDVQQK
 DAhUcmVzb3JpdDEWMBQGA1UEAwwNw4Fkw6FtIExpcHBhaTEnMCUGCSqGSIb3DQEJ
-ARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIGfMA0GCSqGSIb3DQEBAQUAA4GN
-ADCBiQKBgQCksphKSkbE4aCa68r2o7j2xWbxbWP+bjAGwWWYQwnacQ6p5tlhaN10
-ebDAmPVakLo8xxPEXMqWFxCU2AWg0Wtd6TgqIQtIMNXQz6cif5Ufxo3lhus+dLhs
-flz+yTpFD5vREvn0kQ9ce+jVjVzh8bK5qfpaNlaIqQc64WpJKQe+8QIDAQABoxcw
-FTATBgNVHSUEDDAKBggrBgEFBQcDAjANBgkqhkiG9w0BAQsFAAOBgQAYcJ8LyVUB
-5GNqnVdJW4dndeUvllYW3txxuUXhcxgvT7b0glDXSp/cRq0yxZb1jRCLqESsHer0
-o064S5GCWCktZWwbDo75YFE2Vo1R8TChhmD1txFcAi2J161yn9QVoHVbOhyyIHXz
-Yw9zhrnJURZA+1lUpIarcRmkUsbSR25gyg==
+ARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIICIjANBgkqhkiG9w0BAQEFAAOC
+Ag8AMIICCgKCAgEA3msOYI+6e28yBNBmiWxMHGWRjPmGyWoTx5RbYgNbsi8gZ/AN
+15ddi87k8OPC0EOCcFkdk10uh2Fjjc6TW2WHKBrmfesxiTLmCcNbaWmWSauHyIm8
+SQvr3FcUO5Hdy78VJzFHPA6T1J4pRi97xxpDrvvvAVbRemY039pYs51vsVM0wMP0
+zNJ6dwEZcIqZ92faT7s0GJx5AITZGBaC0jDNgjcSTkaPpAUSmys0joZ3vvBQIr7v
+QDEOXNymSY+Rqs/oi7Q32kySDJgVziTYa0D6f3DLAbAslgjuu1N3G1ckWruSaTwv
+XrM/Nu7D3SwOtvRLSM6zWFaBFF2ql4BQNDKs8aF9RhAGY9bvUQVMcgYj5hIhWaTw
+VOzuYoGDJY3U98+ouPR+YpbVWt3GqVcjtr60tPc79i5Yn0shNPnTrykV6oJyFiLO
+loP91MZsWjsEC1wiRYrrkbM/1FjgroWE4uweEOvlDOLhlHy+24dy85lg+wAKE7oH
+Vaid3ibhZx5zVuq+2ewcFUSKsuWC3S3mjgYGUu15t5YafaOhakpvXVoDG1Q8/LPy
+JyFi1f6n/RZXXd/GS9Xgfq3oWNH/PVeuJblNRPNrt3496a1kAtRfVuwGj6sQ6LeF
+/4JNnliS4o2L6ebGo+gqLtue38gf0G7TQvGsCPBq1ujLNNKrfefGzJXCMjcCAwEA
+AaMXMBUwEwYDVR0lBAwwCgYIKwYBBQUHAwIwDQYJKoZIhvcNAQELBQADgYEAACw0
+8lNKnFOQ05UTMVwXi85PIxIST9aRe4faboAew0jw3CvOhUl2HHIqBFFI4N9HH7lT
+cbs+SLzmsdEkRPcEmwxIyVc63aaDwqZsWIa2HUGlys52S3QlsdiU8IdAr2d4w4gB
+1thLOK1SrXYWeOx7A59jsfFaZjpF1R0A7ecsZMk=
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent5-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent5-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent5-csr.pem
@@ -1,12 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB2TCCAUICAQAwdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREw
+MIIE3jCCAsYCAQAwdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREw
 DwYDVQQKDAhUcmVzb3JpdDEWMBQGA1UEAwwNw4Fkw6FtIExpcHBhaTEnMCUGCSqG
-SIb3DQEJARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIGfMA0GCSqGSIb3DQEB
-AQUAA4GNADCBiQKBgQCksphKSkbE4aCa68r2o7j2xWbxbWP+bjAGwWWYQwnacQ6p
-5tlhaN10ebDAmPVakLo8xxPEXMqWFxCU2AWg0Wtd6TgqIQtIMNXQz6cif5Ufxo3l
-hus+dLhsflz+yTpFD5vREvn0kQ9ce+jVjVzh8bK5qfpaNlaIqQc64WpJKQe+8QID
-AQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3b3JkMA0GCSqG
-SIb3DQEBCwUAA4GBABmDywVdbouxznVhI5cnTB2cZTGKMDYCnYx+0pFOZw6ux1eR
-oUF59E/QCIfibOI6b1+Dd7O3hK81aCQxd6oBiWWg8gyCjFcoCVqOkR/Ug176asZv
-72+l6pBLYoZlmPrQXkxtfL+FtLM3/xLdt6hDSZEWyznWcraanDqKx9M4NEgG
+SIb3DQEJARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIICIjANBgkqhkiG9w0B
+AQEFAAOCAg8AMIICCgKCAgEA3msOYI+6e28yBNBmiWxMHGWRjPmGyWoTx5RbYgNb
+si8gZ/AN15ddi87k8OPC0EOCcFkdk10uh2Fjjc6TW2WHKBrmfesxiTLmCcNbaWmW
+SauHyIm8SQvr3FcUO5Hdy78VJzFHPA6T1J4pRi97xxpDrvvvAVbRemY039pYs51v
+sVM0wMP0zNJ6dwEZcIqZ92faT7s0GJx5AITZGBaC0jDNgjcSTkaPpAUSmys0joZ3
+vvBQIr7vQDEOXNymSY+Rqs/oi7Q32kySDJgVziTYa0D6f3DLAbAslgjuu1N3G1ck
+WruSaTwvXrM/Nu7D3SwOtvRLSM6zWFaBFF2ql4BQNDKs8aF9RhAGY9bvUQVMcgYj
+5hIhWaTwVOzuYoGDJY3U98+ouPR+YpbVWt3GqVcjtr60tPc79i5Yn0shNPnTrykV
+6oJyFiLOloP91MZsWjsEC1wiRYrrkbM/1FjgroWE4uweEOvlDOLhlHy+24dy85lg
++wAKE7oHVaid3ibhZx5zVuq+2ewcFUSKsuWC3S3mjgYGUu15t5YafaOhakpvXVoD
+G1Q8/LPyJyFi1f6n/RZXXd/GS9Xgfq3oWNH/PVeuJblNRPNrt3496a1kAtRfVuwG
+j6sQ6LeF/4JNnliS4o2L6ebGo+gqLtue38gf0G7TQvGsCPBq1ujLNNKrfefGzJXC
+MjcCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBwYXNzd29yZDAN
+BgkqhkiG9w0BAQsFAAOCAgEA26zlKDezY8AObq61vclU36w1KIDt7JBSmPKqSSEr
+ZXdjaP/uTq7FNwtyhkCbcMnaW+AbWG6MZZ3HQgH+xePaHT4bqmJqvzFOSc3yTxOn
+NQw5G5ggtD+BJcqkiW71gZMmXXqwuih4q84l02dGY51a2x0DIEHrAi7ERQpHYCmS
+GKU3mEpih5KIJyu6Wm9P7mibMz9OUKM0dWmgcmf14EGf5Z0XSslq3OpPsvz2xody
+Q2fgHtMLTL65yPwKa4EsSlSJrqeGb2Owb1L5yMr2sgbIH9Bqxou0UUxILLBydCjE
+3oWMXOpJbAWUqrjrundF04uLPPNiXBPI+yIX4A714q1Hn429+so8Y+OYuu+i+E8f
+iJym6STBE3O4/18D2CfSeTl7K7/tDZwwKvqmNgLtPuiKyCljZBB0Dx/Shu4xbeKj
+L0mHfcVbwtQD5fZrpXAr1NePCcFkCFE9gzejp3v3rYzX7goDdev/PdIF5VTWb1Jg
+0j6Dt2LAWTiEOGgzhMzkElpAsJy5KD463sKrc0zDPpKIl73fIqtYj8xtthJjb77T
+n07kQlXEW7aZUx6D6GtXexKHWZm4shZ/YKT+762WGHzHyCAVp4NuyxB2aj6TjoAo
+3wxgDJX5uyHvVjflKvvDDbg9hLjpze26xMpMBeFJgtDCODRxk77oWcV/WoHPmMea
+WMk=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent5-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent5-key.pem
+++ node-18.8.0/test/fixtures/keys/agent5-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXAIBAAKBgQCksphKSkbE4aCa68r2o7j2xWbxbWP+bjAGwWWYQwnacQ6p5tlh
-aN10ebDAmPVakLo8xxPEXMqWFxCU2AWg0Wtd6TgqIQtIMNXQz6cif5Ufxo3lhus+
-dLhsflz+yTpFD5vREvn0kQ9ce+jVjVzh8bK5qfpaNlaIqQc64WpJKQe+8QIDAQAB
-AoGASLj7ecI2YXTnP8DiC+nbIEul2vDceFobJsB6pbLkROoq+WaPke2F64pYO5LO
-s8C4G2IkHk6CvadNkQuZ4JrX9xgNdxWRHGAbDedEFnN8gi0inHQjHITj62Il9civ
-JB8cR5fJxzAKZS23elrocrzU6lU90V4gm4VUUJ6dQhZYO2ECQQDYoC6r+Qf9YCNU
-89/RnGdUzL26l1S/GmUw5VfIDorMNbwH1xfg6Z8MukF42Q9kZQdtoh+HVLG7Ljok
-2cLmmBA9AkEAwqIlSsiGSjfJlzQoDi13X2ZgVp6/nicKan1eKCPq+EnWhA5CP1u1
-5WaYBbLjlDl7A7VA7gMd19tSNGHzRQRAxQJBAJe4kRe3wsXOqNBeQnuf7Kty/suK
-JDv4s7jsWG/w53uhgwGGv92yIsiaRzLp7CLns60wqJ5zTkwIU4bt0dkJ1g0CQD9D
-YQe7whqho37oTxS8po51wl6lXvdTDUmr0k0Nz7RAm990mwfpEWitPkCr8tkdDeUY
-pzA2Bx9AhKnOJLqMNVkCQAwJYI4fS7Mec2f2Kv5SInxjMeasGivzDxe99EyLS8jz
-dWfNwtCGA+gmpqcWqpT/JNnJgG4ljseW7Xk6YogngVU=
+MIIJKgIBAAKCAgEA3msOYI+6e28yBNBmiWxMHGWRjPmGyWoTx5RbYgNbsi8gZ/AN
+15ddi87k8OPC0EOCcFkdk10uh2Fjjc6TW2WHKBrmfesxiTLmCcNbaWmWSauHyIm8
+SQvr3FcUO5Hdy78VJzFHPA6T1J4pRi97xxpDrvvvAVbRemY039pYs51vsVM0wMP0
+zNJ6dwEZcIqZ92faT7s0GJx5AITZGBaC0jDNgjcSTkaPpAUSmys0joZ3vvBQIr7v
+QDEOXNymSY+Rqs/oi7Q32kySDJgVziTYa0D6f3DLAbAslgjuu1N3G1ckWruSaTwv
+XrM/Nu7D3SwOtvRLSM6zWFaBFF2ql4BQNDKs8aF9RhAGY9bvUQVMcgYj5hIhWaTw
+VOzuYoGDJY3U98+ouPR+YpbVWt3GqVcjtr60tPc79i5Yn0shNPnTrykV6oJyFiLO
+loP91MZsWjsEC1wiRYrrkbM/1FjgroWE4uweEOvlDOLhlHy+24dy85lg+wAKE7oH
+Vaid3ibhZx5zVuq+2ewcFUSKsuWC3S3mjgYGUu15t5YafaOhakpvXVoDG1Q8/LPy
+JyFi1f6n/RZXXd/GS9Xgfq3oWNH/PVeuJblNRPNrt3496a1kAtRfVuwGj6sQ6LeF
+/4JNnliS4o2L6ebGo+gqLtue38gf0G7TQvGsCPBq1ujLNNKrfefGzJXCMjcCAwEA
+AQKCAgEAyc+nvvQLB/7wCrAxqZWrRcTfGWQ76buncNrrQ/a6F+Zs6AJH/Qup5LZ4
+wx5WYPQwgdJKZLETC+HRHBWJXU+1IC2wYCaszslxMwCgi/gzVmsejFZ1km4Pt+EH
+Sbnj3wBlqCctdI/hkPVli/p/vDvXmAmeZr+o2P+uEEOlYmyCsQpHfnUPRV3houNL
+evM4GuWPj/LTpVUs8cyY0yNx1lhT0Eg6Au5BXVkA9exxBrV6J7UTcUOAAOXCG35W
+H772NF/rp/+BAkGh7217s415FXBjGol0mfaJriUywKROGW+rW+yAP55NeQb5X9Wr
+wJ4+R8FIIirzTj9HZfxLUqYpoAMvKWp8dLrVFGbnqJVUzqqYtvDoXR8TMzNaCjSG
+rssmBYvqpToc7utBF5nvcmtHFTF3kkdbb48T5MkR99nN3sEtSJ7VtxOBcmcAErij
+1djCUqQEv8WCGpEDawwwXBR7B9RRIGhzfBNNj3OJ1Cs4rik3AvYDJ3kxFLWnBkKj
+hLmqttbBQ3/0cvm8O76+ySo77Wq4NipmcbfGF2PkCPEZSrAog1/MmFQ1ZI42liMV
+OO7/GmeXfOVp3cu1kPak0m7OwC+Wq3+WqlGanAJH5WbZc1GF+h6BbPb7QmfgBVOH
+U8FmRnv3opyyj+vwDblYdFuz1W3JIvIaR40boNHL9Cuh8pBIkJECggEBAPs+noRa
+2w+cIkugZMRXwqtnfE9sT4UJRuTODHHh0JrB+y0YT2xvY0ZOuKvM1JZKQoXjnXuh
+iXhDg9THpX9g1nvaeREsp5wclV8hnDSG36hgwBiqp2UdaJDhcvJlpJyUCi8C4rZM
+4lulSVMs48OyO0ujc9l96Gtun/2b/P50hPhaXrTvh9pq1gZqPPfOyN+elWSTS7Y3
+yAJOQ39C/7IK7b9ktdzQfX7q/7YO8wg3uLX2obWqlIypXlhYpx0c5M47TiUQwVZe
+8kcg6Vo31/mb7SAqka3U3yJMT8pH1U04rMj17ia1s+lfrIa6iGjKpGbIpwp4eeEL
+0NTVklXKn+Kode8CggEBAOKgwuuQ4XVejwVQglb+bXeP0Sc1jCODL+C9m5iYRlhp
+s09fhtiM8zWBvtkc5G/rYVN/2IABBm7rwcWVaw9sNAHqXraqHhZVapsePAtr+7Qj
+LZIItUm4yr5phfiUKbsmAhlKuT6xmvWF3suRGxwucdAxnUMvStAJDZWeDLr4mxF8
+ynUr3A3dw+Xf1pe32K7+RpIST5bj0cojohF6ZnDYv7X30b1MTiHmwVdGyFVSB6Ah
+ubKXkLCdmtrRE88MQww7jO9iCzO1zZtB+i42ufgmTE06P21cjVFwdDAwxKgFVcHu
+l9nyGVtvyMFc0aJI5cIkIa7an117Y2ZhaaVBBBAvEDkCggEAQgq8ylfDm3/UOALq
+Wf4jqsCiGZKD+JlurDhJZyoPlo6FAt6QChXhh5qrymUmQHc2s1EYlWXp59X/clAk
+fa4r0xSjnFMbn9qXnRENqXUGl5EPDazWIaswUF0ixPrY4Vu8rYzZoUfmfGafNqDn
+IgrYsVI+4p3JGWxT2+tD/nMMfOfsWy1XjZ9lsPv+GLmfbcCWzoA1d6mEBq8h6PLo
+mZlr01Kax3d6dYE6RFX7pl5XTGWPLlyvfaYLqbdr/igZMOi/6thTqet9Thslm9Rw
+DinhlAH8lzAMk0xx8N8e/5R1PsH+nuhGaYL/q86eisUxn98Aoaz3hKOBemDFfY2s
+p3zWuwKCAQEAn8KkplukFJFsUVrIB3ukv7xqvTFKcV5xygTgQ1pVfM5G0ZWOAeqb
+KQUl/PgZMazFF4/wl4whdOBN4XPL3TtyDkeiLTMPsr7UQDW4LnUZI4wI/AGXizjx
+8/l/sBUCtiO9Jwc+P6wEiUCuJuNMMTmskg/nxQz5uF4pbWfVoZjP+Cv4rVEV99B1
+4P9C0AGShwpJxX+TC9PeeJlWDuwyeyrROv5QpRm9mriJY3VKlJsXNRORTQz7uwXf
+vJHXXQZP3be9SqPbMcDgCKsiKqH5x6137Jk7TWUJWnfj8HTwrWHVark/4yyAI61R
+4L4GKlkpXwsJk7Ia1lO9ZtcVGqISszX6QQKCAQEA7qYgUNPzpK8SvtkS2eS+nMbe
+nP2xC7xT/tFS6YH3TkojjxMdRmNyfHbEAku9L3As3bS4etaMTAL3NGLSPxavlgBz
+Hg3586kl2va+lh18rG89wjdnHdD6TaXN0F9AU6WmAQGG65QtJZjNCZRqR1J5MDSL
+hsPG5lW0aan/ZginNxg9lTfJp1wX50yRAGdsgyzAlf3GiNSpaZ+OT5GYG2G+jyOK
+pkgy62L1fstdaR7yRelZQsrgMNQ9raC9TV75Pg3w/m0GSMsxWLXRT782sENurGOD
+XheXNQcOdD2eWUFO1TFcJj/6PlC90nI6pDGjiKvsaZzTR5UfTa7v6nLcZ74P5w==
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent6-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent6-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent6-cert.pem
@@ -1,17 +1,26 @@
 -----BEGIN CERTIFICATE-----
-MIICbDCCAdWgAwIBAgIJANAIL0WLbvvoMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
+MIID8DCCA1mgAwIBAgIJANAIL0WLbvvpMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
 BAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoMBkpveWVu
 dDEQMA4GA1UECwwHTm9kZS5qczEMMAoGA1UEAwwDY2EzMSAwHgYJKoZIhvcNAQkB
-FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0xODExMTYxODQyMjFaGA8yMjkyMDgzMDE4
-NDIyMVowdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREwDwYDVQQK
+FhFyeUB0aW55Y2xvdWRzLm9yZzAgFw0yMjA5MDMxMTA5MTJaGA8yMjk2MDYxNzEx
+MDkxMlowdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREwDwYDVQQK
 DAhUcmVzb3JpdDEWMBQGA1UEAwwNw4Fkw6FtIExpcHBhaTEnMCUGCSqGSIb3DQEJ
-ARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIGfMA0GCSqGSIb3DQEBAQUAA4GN
-ADCBiQKBgQDBIF2kWViZb+GpjUKfQ2Jevk68mLWXib66z6vCi+mjpcvZeq6A5Z0M
-qNJYftEgSykluxL9EkpRqWr6qCDsSrpazMHG2HB+yip8/lfLWCv/xGAHh9+4XY3s
-UPGIGg+LmvhRCZvgxARxY2uG7AB+WZVMby4TCyAFAT7D/ri4L8iZZwIDAQABMA0G
-CSqGSIb3DQEBCwUAA4GBAFU3MAVxVCmsaoNxr0y+KK/n0iEAzsOH9r0L3PL2gJtl
-p62iOaOPw1/9x8c77RA6z/nXPX9IyAwASv0n8FEdxuIF2+KqFG6bXw5nyfgPIszr
-U7YSV2Pi/Heinr76RrI6aWGtvEuD56Qt3Ce5TYiMnzAWtqEcPLGjgsx0MAv+m48B
+ARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIICIjANBgkqhkiG9w0BAQEFAAOC
+Ag8AMIICCgKCAgEAxT2iTvgZGxgJ78nf0/KYmc55mUpH24UwvQ049KJyYQsMu1Wo
+hHOXrqVGDyXSYffK+ca9wxZilh8sIOTe0CNUHSWY6U84b0owxaNuukrzQHS9avfL
+BY6dlHbdBHBUQIEk13QxslwnwVD55wJXHHiiPO5BJ+ENpFO0o/kVtd8PQoMbTSXO
+XBtJuXX+LHzwf+A46iXC9FqzKXEBUH5EMnFH4oRjTXO0Q/0brhYhT2QKeQG2tbyf
+Zr/SbKlN85ArX/Psm/FkH2Ed30hWtZT1m8s+KlM3NT6LlblUdaPkf5LmzX+t6a2J
+tgYM1Hcc0D9kUEdYzSo/KZMmjsm64YZkSI/ecCLUeKf1wnIDKCGACasaxjrl7xxy
+QP/7KJm7KxZN/9xSeN4he4NkZ9uqvhXrZsjEzLvZtfwJIlIqmqBRAq2vXscEmCU3
+5LOdkP24fzI0FA8+QfiHu3zf2nl0DkrHzCc2YbAyWNRqaLR7udyw2RV8ChUWzcsr
+Qr72ZFRokzi9tdeaH9t03w34vHHA8Wo5Fzmrx2AF555uAryw3hjhL20ENYW7Se4m
+WqY1n/93Ed0jacx0RRTT/SZDWPY+JsqOT55x46lOQfsf0d0RIe213/4/H8rnwFvp
+d5NpPFakjCSM7Nr9B9Znc3M6sIF0sMJZWE5b+P38Awqw7VstSgxiRFtjdl0CAwEA
+ATANBgkqhkiG9w0BAQsFAAOBgQAHi5IR8dV4+xdWU/CZRwjn6h0EvJHyCRNa+Jgw
+7n4+izYa3mXXNJlQzlNvW43AVbqCc+jpqplonUBY7V/NsTHXC5eVo4Zjvs2LVGLA
+CBASVPB5W6NZ74Pjk6HRTnPGWZ3DVmC3+7jzs9zDGVtbJgCnIHcI5wtLn0Mhu6Ai
+ldbmyg==
 -----END CERTIFICATE-----
 -----BEGIN CERTIFICATE-----
 MIIChDCCAe2gAwIBAgIJAOzJuFYnDamnMA0GCSqGSIb3DQEBCwUAMHoxCzAJBgNV
Index: node-18.8.0/test/fixtures/keys/agent6-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent6-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent6-csr.pem
@@ -1,12 +1,29 @@
 -----BEGIN CERTIFICATE REQUEST-----
-MIIB2TCCAUICAQAwdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREw
+MIIE3jCCAsYCAQAwdDELMAkGA1UEBhMCSFUxETAPBgNVBAcMCEJ1ZGFwZXN0MREw
 DwYDVQQKDAhUcmVzb3JpdDEWMBQGA1UEAwwNw4Fkw6FtIExpcHBhaTEnMCUGCSqG
-SIb3DQEJARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIGfMA0GCSqGSIb3DQEB
-AQUAA4GNADCBiQKBgQDBIF2kWViZb+GpjUKfQ2Jevk68mLWXib66z6vCi+mjpcvZ
-eq6A5Z0MqNJYftEgSykluxL9EkpRqWr6qCDsSrpazMHG2HB+yip8/lfLWCv/xGAH
-h9+4XY3sUPGIGg+LmvhRCZvgxARxY2uG7AB+WZVMby4TCyAFAT7D/ri4L8iZZwID
-AQABoCUwIwYJKoZIhvcNAQkHMRYMFEEgY2hhbGxlbmdlIHBhc3N3b3JkMA0GCSqG
-SIb3DQEBCwUAA4GBAAX9mbfgDULEA541c1teuG+eW0KLghFaaotFb0+R6WD1ZQLO
-Url8y1iz6T/qqfuoAWu5OA8/7sKDdta/0mzV6UoGnDOcnWnH5FURmnQPUS/hBJ6A
-mJBslJx6y0z4Rl/fxJUy5K31YbeRHHLEneM211usTv8QguAD0y2BNAQ0Mno0
+SIb3DQEJARYYYWRhbS5saXBwYWlAdHJlc29yaXQuY29tMIICIjANBgkqhkiG9w0B
+AQEFAAOCAg8AMIICCgKCAgEAxT2iTvgZGxgJ78nf0/KYmc55mUpH24UwvQ049KJy
+YQsMu1WohHOXrqVGDyXSYffK+ca9wxZilh8sIOTe0CNUHSWY6U84b0owxaNuukrz
+QHS9avfLBY6dlHbdBHBUQIEk13QxslwnwVD55wJXHHiiPO5BJ+ENpFO0o/kVtd8P
+QoMbTSXOXBtJuXX+LHzwf+A46iXC9FqzKXEBUH5EMnFH4oRjTXO0Q/0brhYhT2QK
+eQG2tbyfZr/SbKlN85ArX/Psm/FkH2Ed30hWtZT1m8s+KlM3NT6LlblUdaPkf5Lm
+zX+t6a2JtgYM1Hcc0D9kUEdYzSo/KZMmjsm64YZkSI/ecCLUeKf1wnIDKCGACasa
+xjrl7xxyQP/7KJm7KxZN/9xSeN4he4NkZ9uqvhXrZsjEzLvZtfwJIlIqmqBRAq2v
+XscEmCU35LOdkP24fzI0FA8+QfiHu3zf2nl0DkrHzCc2YbAyWNRqaLR7udyw2RV8
+ChUWzcsrQr72ZFRokzi9tdeaH9t03w34vHHA8Wo5Fzmrx2AF555uAryw3hjhL20E
+NYW7Se4mWqY1n/93Ed0jacx0RRTT/SZDWPY+JsqOT55x46lOQfsf0d0RIe213/4/
+H8rnwFvpd5NpPFakjCSM7Nr9B9Znc3M6sIF0sMJZWE5b+P38Awqw7VstSgxiRFtj
+dl0CAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBwYXNzd29yZDAN
+BgkqhkiG9w0BAQsFAAOCAgEAKx4x1MMOtJund9zwPS8LbbXNfjwmIw1Xr44MLbfO
+Jm8AIcQdow9MtqEIVOc5YIXycg3a5qtaPUzPbl/BXoXPFwgoinofuJ2JchWH0gr8
+9vo5ioTkmmLUNi4pGdRQYE3aNxZc18y5B0vvaNSdtD/qdEVa0JyOnaFrl69rJPNl
+Idm3+pa1hgioVx6A18vnCdjoRpZhdRqXx6KDnL1BdBxknNMz4dBeuQg0DWd25zan
+salhsjScA9hKbpJHpgTm+XImDUMzDA9NbYo3+y+9wl3hxX7J2OtWGlvn0g7JKjKQ
+IISdnk+Wk2S/u8270XxbWnwX0lFNDwNUZ2AM4++yHLsyV+DcTllAVIyJecu0d740
+KVEySQYuXGn9dtiyrdIT56jupT5v8vQUTd6r3upp6tYN/Txv3ojDVCudZbctEVJW
+xq9v8BfmiKnbcUfACHk3OeScec/ffzrtTlXmZEuk3z90lXZM44SNporYSdflj2KU
+zSgLexPGRLLjTQYlKMUhoqUWFjFu9VK9yAG+6ZBCZYwIUnMDC5DUz0qegdZSMJIV
+rFOLNWKRR1HhH7I6LJ2aOJ9t+b7DDnm8GvkudegVEEzS6jw7fyBD2Bv2Qt81ykXd
+ac460mHpyRLJToxd+KDpcYv5alTJjTWGH4uphlMbRaq5TC5X8xqE2lA7IMxcs2Ne
+Z04=
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent6-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent6-key.pem
+++ node-18.8.0/test/fixtures/keys/agent6-key.pem
@@ -1,15 +1,51 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIICXAIBAAKBgQDBIF2kWViZb+GpjUKfQ2Jevk68mLWXib66z6vCi+mjpcvZeq6A
-5Z0MqNJYftEgSykluxL9EkpRqWr6qCDsSrpazMHG2HB+yip8/lfLWCv/xGAHh9+4
-XY3sUPGIGg+LmvhRCZvgxARxY2uG7AB+WZVMby4TCyAFAT7D/ri4L8iZZwIDAQAB
-AoGAIfEMRBwnxB+zq1bWRKNVII2VzPORxqZAzRg+eZyZXVeAMiKrlJ/GMDljboYr
-Pt+2xZjRR4T1ZtC9qnvt/VlM0uWTEIgyzo29ZO0Bd9yMnIF2EUlzVtW07UnN6+VW
-z1/RxOgBiAvkrSTgN4SOQJJIgOZYAt2Xhrkz/0CLwEOis1ECQQD+1hQkEYwvNH6Y
-qUVoWvlNg0Q4kozNQCrwUrkHCtIOxCmr/KxcwcPBaVmEypcCnwf78KbgUQ/5oIgZ
-OReuNcWjAkEAwgIk38VZRxSyBP1RThbKrK5h+GIwkEO7RW7lZ9yoVKhV6YDMWof0
-xCl23YwdpflaTUHuBVOYa3EPegkOGCfxbQJAdCTDpzCsMHN/Yzp6nLYhu3chJ5t7
-OqyNJVy+YXxIAlzbFTyind/dxQ+rsf7XVmV+sQ+cLs4jNsU4Yi6IIWj2ewJBAKRH
-OH4bF9vulEdRUTV0ay4Jg3/VdRXTpJHIs4xc9lSpLgZJP8Ew+nvYOISlDr3qBSMC
-PtBX1uqzk81cOYkO2YkCQBUVMew70XetUXgh/2KOWyG/87uYy/s/NZ/LGImvo+tq
-FUapBPapob9I7WA6gRYVseiE+mSGPAciGIFg/d6iyxI=
+MIIJKQIBAAKCAgEAxT2iTvgZGxgJ78nf0/KYmc55mUpH24UwvQ049KJyYQsMu1Wo
+hHOXrqVGDyXSYffK+ca9wxZilh8sIOTe0CNUHSWY6U84b0owxaNuukrzQHS9avfL
+BY6dlHbdBHBUQIEk13QxslwnwVD55wJXHHiiPO5BJ+ENpFO0o/kVtd8PQoMbTSXO
+XBtJuXX+LHzwf+A46iXC9FqzKXEBUH5EMnFH4oRjTXO0Q/0brhYhT2QKeQG2tbyf
+Zr/SbKlN85ArX/Psm/FkH2Ed30hWtZT1m8s+KlM3NT6LlblUdaPkf5LmzX+t6a2J
+tgYM1Hcc0D9kUEdYzSo/KZMmjsm64YZkSI/ecCLUeKf1wnIDKCGACasaxjrl7xxy
+QP/7KJm7KxZN/9xSeN4he4NkZ9uqvhXrZsjEzLvZtfwJIlIqmqBRAq2vXscEmCU3
+5LOdkP24fzI0FA8+QfiHu3zf2nl0DkrHzCc2YbAyWNRqaLR7udyw2RV8ChUWzcsr
+Qr72ZFRokzi9tdeaH9t03w34vHHA8Wo5Fzmrx2AF555uAryw3hjhL20ENYW7Se4m
+WqY1n/93Ed0jacx0RRTT/SZDWPY+JsqOT55x46lOQfsf0d0RIe213/4/H8rnwFvp
+d5NpPFakjCSM7Nr9B9Znc3M6sIF0sMJZWE5b+P38Awqw7VstSgxiRFtjdl0CAwEA
+AQKCAgEAuW2UJYGAuMQEs3WqA2vSouZF0yLOI0Y358OChBokRa9JrOR+pOa29u1R
+y/PWsVtvaxq4U6u7yz5LEjGYdzx/W1d5Rf3AJIAvEd7vvhuwTCxz1xhVuUe2RVjp
+t76StoGyzQrjnbadeVRTXCYLFy0h+P5UKtWdKZu3+1+0VWL6yaDQDds5AcZ1rPGM
+nu5mexGk8BvICEksng0rGy1Zl8J/viiP4sSf+Dr/xs3vSt/WKKS/zGMjyEkV3inc
+qtoirPZ4vrJVWlrG5p7vRY5gyLJuIwRQprHCMGQnTuFhc737Vb5vFXycNo/BiAzi
+/bRa7k3k4hyRfNhmK17jXhomNUFr735NX5Jg/A/hzNcjggWGwAM5qfQCkkV051sJ
+NNxxyXr5t3rbVAouoMRNUbfcyK7+Ioo4vsPROm058iITqFUvz++LPvXV1cOfNrK2
+awK3DW0YGZtDs6Iy86anwiPN6INTklUxqQYxxj5P6WpOVNutpDacsKhLBM2BsnaZ
+JkxwP2xSwMMiDj8R1IWQRHPYNAKewO7vdSIW1TGwzlrP5kVKAawKXPAG82SMRYJw
+SqETtTXvqjdBT5KMWsCIMq5W2P/ZbVJac8n9cg7nvBGK4rHP4gzlarWMZe3w+F7i
+HH+MtdsazndfL4FqxrZj/pBKfoq/0+bXisN/j8i8ujjli74MGIECggEBAP5ccOrE
+OcOE1UxlsbIiM/I3TBqor+xRIwFd2quiz09r/+IoR+0SpLEhkt2T/sOy4yXd41ia
+LUkVu30poo+2W24QjtsTEEvS1ZHX1hDQMgqYKD0HJx8epFOzbQWbL2O0iouzDzxQ
+ScZ7GbjpJCwf0jfMwfaVKeDVsFnVUHXwuc2OPTm0XUInvdVfEvuoRjFzLlISkumZ
+JipHj9Pzu3l1jmkhTw3d4VHHGH11LMFkoW9MwyDJ1AgrMnDNB2vSsHbSSz0fG+YD
+x0kRiF4hfvO/hm7N23gOvY4hSbD0fSgLjmfHjDCl9zrPTFVoHzuwY67jwtypdulP
+L1E6syKpnJOaZdkCggEBAMaC+aErbJcI4fYPJ0mxGTKvz9dWXPB3U7oI2du9us9m
+mom3BlMDa13M3vIC38CRERcoiyLgbh6eo3zP4IVQ7c95kZY+gm3JqKY1ql9KpyYm
+EHpHhvZmZDH4gOmhXhWCtuEzyiKJQgOzCzsQ431NZLzcDqKYIwl6s4MgvoqH7a1U
+i0TAJvqij0IqN12kxyXkpsJTxvrisUp4mJiZv9nC+i/yemi8Ispz59wyiaRepGyf
+wPK2SJ0Biqun7CLgnvGyllzT/rlgu256taCA1TI+O41uk0oa7rJeJIqerSuAiDeb
+OZcKRP05x5bBeYZ9SF3cK6o90L4ImEsUbrO9dLmb7iUCggEAYfUJy2/Max7B4k0b
+tOV1Keff2Dh0n8bTe1hAuW7W9dg60EcBBCz03f53B/xR1d4TmmasglfF7up+rQIW
+x8/LwESCL55U3U6RzzvXBdyYGPpmLLKPf22WjWGBL/KGupgx7wEDcg/diaGo00x+
+ELwkgjbJ0AO8AAlrg/0honoVpDMOIyXx8GNLuVuG/QHeHnckNJlQmMPRLeEnY9Hu
+tiTutlAMmOOFPbA2H0K7/LKbdTiTwRHZJVg1cbmUDRMzwF7WePB9FwUHf31QDihK
+Mubkh9GhnsfWemEZdxuBA6YHrrHrrGSOJMjmtdGGjb8tdJHkXoSo9cr4BwlbKZM2
+aKXCaQKCAQEAwMM4UifPp44JlZ07YlSlilGsuobzctH8rJ6hpr7DI1iX4finHH8q
+iAhDya++xMhZVyqYiUJzzCnxyWY2uF5QvvJmzAVhM+jpapScHzJnwySdRZAlXyko
+TFnqHHQE31Bx85cNxqGQR/VZh36r2OYJV1c8ouimxTKwFDn8rJq30mNRZW5Nq32l
+LPO/laBHfeVdhUBC7SaqUA1w6GYMJRIvTdXgxclC42LPmEQ+4izA73XlsAoP0t5H
+YM6L/4Ys8bigR/3X0P08pFhl/Dh+Te6ed4UTY5dpiCzgwPa6Y47JN23061UejuLH
+i/3N2ynW1sXQZNzn0VCZSfbKVKE5Mm9XPQKCAQB6DSgo61K/em9hZdaRzTxeA/vj
+gAXp4oT/SFcbTbX+oijnZhKbDqbOHw19QN8Yh8YYovbkJtKzgUpKsQwVTCcWUF2a
+dPpBh5C6iArprwcxTsPVZ9wyAtKdz8E86lqBZipnYvVWWblmwe2EMT9s2nt3Em26
+guzw1PCViI61BA5T//6nMy2Ro/CvIT4Ybc5Ccmw6HVCs0I0HbIYc1ztF9dS1DPKC
+O4bamds3kp0gcLGPzZrzn9UzOgIR0rp42BATFQFM/WJxwVkKjXYV2usJu7bHSg0A
+IMbKvdGMJV7fcs/GVvaX3C66medoaL8niSHykBn1eTaj513oXjbJ3p0ZAmGc
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent7-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent7-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent7-cert.pem
@@ -1,19 +1,19 @@
 -----BEGIN CERTIFICATE-----
-MIIDEjCCAfqgAwIBAgIJAJ4TtCDh9ccYMA0GCSqGSIb3DQEBCwUAMDIxCzAJBgNV
-BAYTAkNOMQ4wDAYDVQQKDAVDTk5JQzETMBEGA1UEAwwKQ05OSUMgUk9PVDAgFw0x
-ODExMTYxODQyMjFaGA8yMjkyMDgzMDE4NDIyMVowXTELMAkGA1UEBhMCVVMxCzAJ
+MIIDEjCCAfqgAwIBAgIJAJ4TtCDh9ccZMA0GCSqGSIb3DQEBCwUAMDIxCzAJBgNV
+BAYTAkNOMQ4wDAYDVQQKDAVDTk5JQzETMBEGA1UEAwwKQ05OSUMgUk9PVDAgFw0y
+MjA5MDMxMTA5MTJaGA8yMjk2MDYxNzExMDkxMlowXTELMAkGA1UEBhMCVVMxCzAJ
 BgNVBAgMAkNBMQswCQYDVQQHDAJTRjENMAsGA1UECgwESU9KUzERMA8GA1UECwwI
 aW9qcy5vcmcxEjAQBgNVBAMMCWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQAD
-ggEPADCCAQoCggEBAM2+DkRxPeRGI4JK0YpuaqCJkNswMYMyZM4VQYyn99SL7xS8
-lB9P0vIm+K1P4198WXwUHSykWRcyy54nMNpq+9Dfy8BalHRaUa8BO/7UQgipRGi0
-HidDk/bAuNNHNIzJr2sGYGHsZuHkO9inEqDcqrSlTc0G0zyLry5LekRZRTgAlXpl
-C9PsAZl0J+gGA83rUhdD/RDjaT4ldqjLKycHvcMLCIS6Wq0TszYCUdbvMsageDcQ
-zQerIJkHzfJFGYCetQR5/fiIKyF1bICKD22AnpLlfhdthAhLVlYQn34IlYJwdptq
-2miktcqvBw4rnBhE1ONj8DqF61S9BKr9aCf5OoMCAwEAATANBgkqhkiG9w0BAQsF
-AAOCAQEAbek4GyLi+vVeUg3Od17lr8qT108iP3esUJ5cfPrXaexxcypRAmYPB7r4
-NA/vVeOTNkxbb07Ct8dmz+rTn+FflI9h5eKRC7hbH/rFTDEfnoS66eSlxr/jJgtv
-LWuKTMJhzXjgliHleaBDGzo3mR5hcJbQvj9qyK4pXjxlt2QvkPdx2H9H76+nBh1g
-TY5bW4+3NFaHfaR2p2T20bY3no25/vfV7K5endff6pgzcZR3/SptGTywC4EZzIcz
-9Q0JnALQtxAAxq1yrljQcvpjM/aAYY7BxwFHJuLmb/FpMulkzZ2vJALluF/3G5ne
-RT9QhxJdwUz+Juv5QKH2i+nnb2Ur6g==
+ggEPADCCAQoCggEBANE8b9BsxopW22PEetz3St3bkhOYNcwB3PZHOlI2cDB9qNKJ
+DjCPjBxt1fiG01sopOtRGk+nvj+4l0oc+R5X6eEpGGrfDKP9W2x7woBxRLPa0Xra
+u4DuAjBy8SfsiuOJlnzR8D60nvH64lJcYeBLSQRocGZ+LqKjYfE1h+r8oStKiOT9
+m5IWHBZxC9mc5qylRdcH3eEoRekRKzrBgRJE38fKX0seM2r9D62SOcjd2oZHLoju
+D4ly919AFPT8vYO8Z+thPANomriLoSOrexgKRlpnIQmGTKhTmWIlkOHCc7mwzf/o
+L88FaNaVDf4EYk49GcbJpIcR9Bd1bkdQsMOOij0CAwEAATANBgkqhkiG9w0BAQsF
+AAOCAQEAKVuF2GTgTpEk+OEoG/0ShG72foquVHMiMd8BO40UZ4HJRuW7rAcseeeD
+MD7869UJpdBTrmNXs7YQj7wHrPR/1esigpHccjzYcJUV4NtOPLB8XlObLs3aSMAS
+mDZLhOuxTpKzsjs6cppX8sIm+4tL2SYHWJBSEdQKAQOSx6qiPU0DW/Mw5/9oW3aD
+JIq7AkPKu72GxCDo6tfUdLrkhFBKFt6lGMFIBtlMeZJjRgQQipCmv/WbHR20hMQS
+kcfa7MSeOGOZhcepdhGe2b9voFM6yqlpqqa+GhkkMlKmlvTqXIU+hRgpBYELu29f
+Zt9Kjwf1jpO8GMXgLn0SrVeH135uyQ==
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent7-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent7-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent7-csr.pem
@@ -1,17 +1,17 @@
 -----BEGIN CERTIFICATE REQUEST-----
 MIICxzCCAa8CAQAwXTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjENMAsGA1UECgwESU9KUzERMA8GA1UECwwIaW9qcy5vcmcxEjAQBgNVBAMM
-CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAM2+DkRx
-PeRGI4JK0YpuaqCJkNswMYMyZM4VQYyn99SL7xS8lB9P0vIm+K1P4198WXwUHSyk
-WRcyy54nMNpq+9Dfy8BalHRaUa8BO/7UQgipRGi0HidDk/bAuNNHNIzJr2sGYGHs
-ZuHkO9inEqDcqrSlTc0G0zyLry5LekRZRTgAlXplC9PsAZl0J+gGA83rUhdD/RDj
-aT4ldqjLKycHvcMLCIS6Wq0TszYCUdbvMsageDcQzQerIJkHzfJFGYCetQR5/fiI
-KyF1bICKD22AnpLlfhdthAhLVlYQn34IlYJwdptq2miktcqvBw4rnBhE1ONj8DqF
-61S9BKr9aCf5OoMCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
-YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAKcwglrFyAj+pc26WqHv5R9NToUKF
-1Yd5zkExZHWH5glrAprCdRhUY575KcY1Sz4KCCRADdYo7KGUFHi4B/N+iyIS9m3t
-TWpJQVq4o98hF0+FalhdYyIND2FdiTmdxzmi788JFcTKZT1ryKyoB7vAj0kvXdED
-3VU2mDoxPc17ZInR5x0A8hJHDHY9SlDL96n6QTEAByXfqNq/c8S7bkBPEJJUln7G
-L/8YWxQJ25971PEX/QLbWADMkSPGkHCHF0znZhtJ6wxTFRkdQJSa9FASKpVgDMMu
-wQVEnOa10z2aQ3PayZUHh43zq441FakE7LAseeOoJChPb00lFrN3ph+TnQ==
+CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBANE8b9Bs
+xopW22PEetz3St3bkhOYNcwB3PZHOlI2cDB9qNKJDjCPjBxt1fiG01sopOtRGk+n
+vj+4l0oc+R5X6eEpGGrfDKP9W2x7woBxRLPa0Xrau4DuAjBy8SfsiuOJlnzR8D60
+nvH64lJcYeBLSQRocGZ+LqKjYfE1h+r8oStKiOT9m5IWHBZxC9mc5qylRdcH3eEo
+RekRKzrBgRJE38fKX0seM2r9D62SOcjd2oZHLojuD4ly919AFPT8vYO8Z+thPANo
+mriLoSOrexgKRlpnIQmGTKhTmWIlkOHCc7mwzf/oL88FaNaVDf4EYk49GcbJpIcR
+9Bd1bkdQsMOOij0CAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAO6TxDrT3i4Z8MpouaeClU/k5cpZ9
+cypmjq4EANLiX62bwlOPE5Y7Ms/LzmBrsgBE572KVUiy9EGpDApRC6Me/jqCp38v
+4vgGEldDZNiB51zm2Gk+X70DfRtsloDqsT4jnUnO5hZHRvGCGLGZPSo3V7a9eSU2
+ydnnPMfT6f9exhAEbKkhupxwanhT7gVfM3Z5f+l39f3mVwJmlGe3IYpi710/qy4K
+P03GstYgAmnmp6GhdNz9wbwX0A8+qI6lCf0+dj67G3RUSwo1rvjImep4NzBonscY
+U2peOXktduVgIBZpqZnlOfiBRlNP+uVGeaI4u0lwnm7uYZlF4mpB8xr1CA==
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent7-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent7-key.pem
+++ node-18.8.0/test/fixtures/keys/agent7-key.pem
@@ -1,27 +1,27 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIIEowIBAAKCAQEAzb4ORHE95EYjgkrRim5qoImQ2zAxgzJkzhVBjKf31IvvFLyU
-H0/S8ib4rU/jX3xZfBQdLKRZFzLLnicw2mr70N/LwFqUdFpRrwE7/tRCCKlEaLQe
-J0OT9sC400c0jMmvawZgYexm4eQ72KcSoNyqtKVNzQbTPIuvLkt6RFlFOACVemUL
-0+wBmXQn6AYDzetSF0P9EONpPiV2qMsrJwe9wwsIhLparROzNgJR1u8yxqB4NxDN
-B6sgmQfN8kUZgJ61BHn9+IgrIXVsgIoPbYCekuV+F22ECEtWVhCffgiVgnB2m2ra
-aKS1yq8HDiucGETU42PwOoXrVL0Eqv1oJ/k6gwIDAQABAoIBAEvZcmcXHIsotHSX
-YrLXTCYNMUMtfENy86jqOzVAw2Qvhp+teiolAo7VgT5bwmZ0cIUG4U6Q9GtSBbEz
-n5YWdOmnZ/VtL2fJ2G1dViH3XLTWumqjZK5zAnyoxjrV+HCi9jHNswDG55MF0m5o
-Ab0ePSzF+G3Kw1uB376Agv3pr1QacVmMxsYxuUm8Ks0H3hB1E1rYByOcFgljq0EA
-E7GuYG1JyjFcGsoyNPykpJ8Ri6mko5sbE7ndaTPqiYkovl02nsqGdiXSyrlC1Q7O
-+XjfOO0gig6LFsWiKCWxQzJUOTD7RsqRaZTHnEHEf9iIcFmHh52i9Bt5r/lqNA/Q
-D7V7vsECgYEA9pYzzZR4+xSYKfQHEpIdUGQTh1kFIdxHp/W4HD+3NJc+DIeUWGVl
-7oUGcZvcEDfQojM3LUNAof/NuPuLinjKKzycRBU05ApuLIP9R1mq8ndxe98jxOIM
-sCd/UfKHKAtOlyWnLvLuJZLozVYack+9/ZnEDUuz9u1S3l+IPUNiSmMCgYEA1Ziy
-jFvchUecrXz1PFpW95psYMkCcgDYg26UwF8jrEcf26DblXs3O8mPGUqCwHOJHk+6
-fEXMGbF7ocr+b2HMKuq5EaCFUJqu5orZLDqFuxCDee2OS/pxfTXaXwrnCKVrYzJU
-9HNmuac4pryWnarnGbA56BlXp9mJsAbRcxiZOWECgYEAm1CKOn+9H/Cd0zb4SXMs
-8ZjHUCX6/JPhsmIr3+cl/wMQOxYekvrzFCRHpcFVAAYX7EI0C9djW2Zi7pPKFaL1
-O/yGNL/iu4vyTymnm4xYBzbCjRJEVltHQKDwKe6HwOo2Sy+VORYceCArcEI+kCe3
-9IconHNFXE+pNZWYm3XY8B8CgYBmEHUhBLQ3K6T+cXttv21XG382MFbuyuCqzShf
-VBbjt4jNlevXXe1isEmkuCoKdCrNRSPDRkbk8B43jZxO9NhumYKdnaqWfZOdrjNg
-IwbMAHQSyyT3wVCBmD4ktDz5sLHD0MUvmgU4KWO0qOD/ri6H4+GHurRcDGLyrg9f
-hB2TgQKBgH1WLbZEHvY07coCUdAywMCjcR2zmKrxo2rsmVrfjNNF0X0mh6Tsw8Af
-BpL/j2bb7bHIIVKEystD2lx+zmOyLZOmT7nvZ7nFKiRKe4HwHiZW5N3JJjpGfBWU
-vzPAbJHWnyctRihxgXbq6eGJEv5Dwgf6ERP83Cnn3JiDUNPuEIa4
+MIIEpAIBAAKCAQEA0Txv0GzGilbbY8R63PdK3duSE5g1zAHc9kc6UjZwMH2o0okO
+MI+MHG3V+IbTWyik61EaT6e+P7iXShz5Hlfp4SkYat8Mo/1bbHvCgHFEs9rRetq7
+gO4CMHLxJ+yK44mWfNHwPrSe8friUlxh4EtJBGhwZn4uoqNh8TWH6vyhK0qI5P2b
+khYcFnEL2ZzmrKVF1wfd4ShF6RErOsGBEkTfx8pfSx4zav0PrZI5yN3ahkcuiO4P
+iXL3X0AU9Py9g7xn62E8A2iauIuhI6t7GApGWmchCYZMqFOZYiWQ4cJzubDN/+gv
+zwVo1pUN/gRiTj0ZxsmkhxH0F3VuR1Cww46KPQIDAQABAoIBAGA550J3hCeDIoqT
+hd2xbSDzJg8U5UjZYoFw6IV+u8HRSJ7qoTf8DHfWm50chH9cBdyjA/K2NmyfvBQ9
+luQ5kjoW6D/eLLYOgsk1cZIYhXwhmk1h4vG8hA08D2RJCOb0E917v31XG/I1a3ht
+PCYm+YGpdmfsiA4MDbMc51FOteBuJeQ/lrVfjoexDmE4Rhxf4/p4LedgkxIaGO1H
+XWOUJ07hAJo31uKU+mfVONkHFTXlp16F5wd8uy0GWmwdsQiQpxDyiPMtpR+T9Fr+
+epecLGyfPlfR9RM79XSCu13Ze9TeQYP7DLkaBS9X6XQTmfwa+yOVdxVowsSEeCZk
+v0gUWwECgYEA7rB5qGHASrdYQhGKunHrDYrie/SbtOmZ+DlRhLQOxSeFY/Dv5X5o
++iAaI2+U0LrsRLT6hT2z7rv9+NiTipZ1qbojMRKlXGLYZN14At3fQ8CGu/gg5iFe
+i6DQ07x9ZhqZEkV4mSK71jxcS4olCsYMQKIjbO+BIQgF3/z9NApBwnECgYEA4Gkh
+OmRMIsgMZ18ISrMD386bnJF5we7O3Mhs3xc5lDIwQvpM31DOI4Jlf0/OkvnpBOeC
+j5WKnzun+I1o/D4FJAkAqwyQ4+rfz5kKnlRFaITH8RYDnQvs/vAuPxgdQSQnejS0
+0r03rrjS3Mh+odZbzHuV66sAKIet03BNZDiSko0CgYEAnghL1Dc8OHvhvISexEju
+Zp4AzC8s9IACCD3zAhiw+snq4DvCqcCjQpAEnlefQP+xtN/aSmOLAVKImbzTGtv5
+dXGfITuGLn/nZAjaD5PU4afTzRpAtf17+ph0CuxuXi7YECfc7A76LKL0PLNvSQ3N
+SPHQ4KxGY5dLRHzdLBpqZZECgYEAzhL8dnigCdiWu15jaOVXC3DZKT/WBabcw5Yx
+k/Pp5hLoirj7Wp8DjUwzFGQfGIzx1JUcKepzL6JCr1Z8MBZr9hlmSazyaLnI6QRq
+4dYVHtPAH73Nd6v56Gp6rWnhg25F6kz4OAWWWpZS3HgpAUitZ+udUprnPFDL6TxZ
+dU0WHAkCgYAKhOuPyPb+T+pyraE3JckMn5zly+qQEdK8LzOMESidzVKGu9K4t+ed
+FkP42WpmBEUgPq8qf791tgKW9grmGsDLeWf/nDbZ2l+BCzo2OmslaYIhlt245rWj
+bDisN2BmDeRXXjWfTwDoNy1UO5UxcmokS6UYLv3NRgkRzxEhuvYAdg==
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent8-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent8-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent8-cert.pem
@@ -1,20 +1,20 @@
 -----BEGIN CERTIFICATE-----
-MIIDUDCCAjgCAQEwDQYJKoZIhvcNAQELBQAwfTELMAkGA1UEBhMCSUwxFjAUBgNV
+MIIDUDCCAjgCAQMwDQYJKoZIhvcNAQELBQAwfTELMAkGA1UEBhMCSUwxFjAUBgNV
 BAoMDVN0YXJ0Q29tIEx0ZC4xKzApBgNVBAsMIlNlY3VyZSBEaWdpdGFsIENlcnRp
 ZmljYXRlIFNpZ25pbmcxKTAnBgNVBAMMIFN0YXJ0Q29tIENlcnRpZmljYXRpb24g
-QXV0aG9yaXR5MCAXDTE2MTAyMDIzNTk1OVoYDzIyOTIwODMwMTg0MjIxWjBdMQsw
+QXV0aG9yaXR5MCAXDTE2MTAyMDIzNTk1OVoYDzIyOTYwNjE3MTEwOTEzWjBdMQsw
 CQYDVQQGEwJVUzELMAkGA1UECAwCQ0ExCzAJBgNVBAcMAlNGMQ8wDQYDVQQKDAZO
 T0RFSlMxDzANBgNVBAsMBmFnZW50ODESMBAGA1UEAwwJbG9jYWxob3N0MIIBIjAN
-BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvCbXGqz553XQ+W9zsQEaBc1/mhd4
-TFjivwbK1hSdTuB8vWyOw6oZuqAJjctcIPmNXf01zV1+cAurpoU8k9SmtetwqaDV
-0K5ooKUuzgAefRoLJqU0XonW4VaK0ICQATkxSWdJzYET68NTukv5f9Fh0Jfi2Q6Y
-PKlgUIuoTPQJSErAMsdph4KWMP7zsaEZNJhmZ1Lprfm4DdVnwUfYvDhq5VmAHFLj
-Vor/z3DJS+pW9oORDta3CMvAY5oGcIYWWMxsoG9B9NtTTs58jjeFpJrw/RYJA/CM
-uRawLWKt/z1zPhzmvknTKfAIc6SjbBqu8Nx/Xvcd61c2V39U/nZDTs+H9QIDAQAB
-MA0GCSqGSIb3DQEBCwUAA4IBAQBfy91+ceZDfZ0DnHHAlm8e+26V5sdrdOXZJtkc
-AacDcCX6AD1iMK+0axBgG6ZJs6m87cmFdaq23pLpBLQ+KHSdG5YgRCEuWW+RaJGj
-/vVn9AS4eB3EmX0RhhJgYyVbN7ye8qjfAv0NtHzUsdMS8ay3HbdUCtrcsHonGDR3
-t/0BGsYny9Kt2f2PNN32UEkx/jhcssXwnNGxyxR/04heJUe6LI5ErdQoxxvaZtrd
-u9ZgjSxix4dFH4nTYEYe3oXM1U7PakbzOzJvRMmDh8vYyK7/ih0w8/DcsK0d1Oej
-mgtTF/IyJqy8T9goFf9U2uSshia+sKJBfrrzRaUHZMx+ZobA
+BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA4r+Hwd0sNf4wOlwefgQWG8SyU5hp
+xX2xwZ+Y6TRCRyTxJajdt1cdq8XZmMFgL7Mc/s44oBRcxkxGSiHz8JRa+xoBHFRo
+k4Wg+EeWpn8EUk3RX2Qn7rovkPAuTdOcGOSv60RmrB9ZAI0bE0b353s67ptsfb8O
+XjlManINH8T2AMFXqxqPiYe0Fqfor9NYm8AGdh/G/NPkZJR4IdCW/KVvTEqYeDjK
+zbnRFGsaGSNZj7xWYK/E7kHxtWd0ipvs/pYuYFsbguUjsghUFh+0ZtjZ7VVB4A2b
+3m1eZsDlmSc83g9EbYDOldAldH6y3INJ4CUKrhdCAhqAjAAmqcMxMAIWtwIDAQAB
+MA0GCSqGSIb3DQEBCwUAA4IBAQCTCeP1IkdmBFhokPe79eUbw0MBPm/mGTaZ+C/h
+gXHAXMJGRytg/ki99UuW1MlqRePvTTSpw0/046cUXkeiTjYRNi7eK98bsXyFUQfc
+opvQvLHBQmLeeRenZKcV8uiwWGAOk6qf2WevDVJfiIqAfMKzjr7CQHALHgw82WGp
+jbtQV8IqkKsB15Mu5mFtBjUDi3sbBxOx+MpO9Bknnq46Dabep2lu7EfxkGhwJ26Q
+9i64SPQr63zqzQ1iVqd2O4Zxr8tFtQXnS4TB0Z7cgxO23MgJAiBnBjko/CUzx0og
+/1G8wPn0U/Vp7ebdjXu8rBrI61HM627KGD6K3TZwxFpDr3H1
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent8-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent8-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent8-csr.pem
@@ -1,17 +1,17 @@
 -----BEGIN CERTIFICATE REQUEST-----
 MIICxzCCAa8CAQAwXTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGTk9ERUpTMQ8wDQYDVQQLDAZhZ2VudDgxEjAQBgNVBAMM
-CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALwm1xqs
-+ed10Plvc7EBGgXNf5oXeExY4r8GytYUnU7gfL1sjsOqGbqgCY3LXCD5jV39Nc1d
-fnALq6aFPJPUprXrcKmg1dCuaKClLs4AHn0aCyalNF6J1uFWitCAkAE5MUlnSc2B
-E+vDU7pL+X/RYdCX4tkOmDypYFCLqEz0CUhKwDLHaYeCljD+87GhGTSYZmdS6a35
-uA3VZ8FH2Lw4auVZgBxS41aK/89wyUvqVvaDkQ7WtwjLwGOaBnCGFljMbKBvQfTb
-U07OfI43haSa8P0WCQPwjLkWsC1irf89cz4c5r5J0ynwCHOko2warvDcf173HetX
-Nld/VP52Q07Ph/UCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
-YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAsLb6+50b2ck7DOop0M+EitLaY3H2
-PWZBS6A86RU+5y30pJsFKCtefN8Hb21DwSwqhkcFukHzZRcKuGA8IsMhlE2u7YAU
-Bx8pX5jOBhFWbUG+mAoe563XPCQNZ3GbKg3pGOCJ8b6gflmGvIxXlzQlR8lg1RG2
-dT5q/sWTOXOsDyu49bObDw0jEFM/HgHzpFyHdrnh3P2vEULx7qdRVUXQ9JIsuPjB
-bys9FhjDmV9yEabWfHRXqrFY318CPit25Q6Cl9G4EFMCYkUX2nVzjLojExkwJHdf
-y4wDaEzxtqJgEEaQwMu+j68v3wgYAGk0yKMFNDQ0gaSZkAQ6u8I5unTGYQ==
+CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAOK/h8Hd
+LDX+MDpcHn4EFhvEslOYacV9scGfmOk0Qkck8SWo3bdXHavF2ZjBYC+zHP7OOKAU
+XMZMRkoh8/CUWvsaARxUaJOFoPhHlqZ/BFJN0V9kJ+66L5DwLk3TnBjkr+tEZqwf
+WQCNGxNG9+d7Ou6bbH2/Dl45TGpyDR/E9gDBV6saj4mHtBan6K/TWJvABnYfxvzT
+5GSUeCHQlvylb0xKmHg4ys250RRrGhkjWY+8VmCvxO5B8bVndIqb7P6WLmBbG4Ll
+I7IIVBYftGbY2e1VQeANm95tXmbA5ZknPN4PRG2AzpXQJXR+styDSeAlCq4XQgIa
+gIwAJqnDMTACFrcCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAHhGZ6bAtjOqtHK6PFa2OA1uG4GFj
+X20xtUHV8jkMa8i4XLUv5JHM8ly+3kWnpRPIip/QgNhMT9Th9OIu4bMRTvImUuQB
+GDs3/DqaKciSz4bvJ46J5taP+XKPNhRJU/VYA66i4bbn8IcMZkllUWZKFq/8Et0o
+VkzncfQl7c6F8GxQ2V06B6I+gh0FtEhdm4Z9noI5TA0NjYNMM9JqIjWLC3YsXFjS
+0RdmIVugSclOOVq1S5tPhew0Iueskz+tbApT8gubo8B9BVXNrLkFfw7FZWH1Q+Wx
+R5Dbe6uXc60f64VaVnz1E9i1K+eNosMBwOSMSj7i/jPhRyx1tnKxbd+Yyw==
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent8-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent8-key.pem
+++ node-18.8.0/test/fixtures/keys/agent8-key.pem
@@ -1,27 +1,27 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIIEpAIBAAKCAQEAvCbXGqz553XQ+W9zsQEaBc1/mhd4TFjivwbK1hSdTuB8vWyO
-w6oZuqAJjctcIPmNXf01zV1+cAurpoU8k9SmtetwqaDV0K5ooKUuzgAefRoLJqU0
-XonW4VaK0ICQATkxSWdJzYET68NTukv5f9Fh0Jfi2Q6YPKlgUIuoTPQJSErAMsdp
-h4KWMP7zsaEZNJhmZ1Lprfm4DdVnwUfYvDhq5VmAHFLjVor/z3DJS+pW9oORDta3
-CMvAY5oGcIYWWMxsoG9B9NtTTs58jjeFpJrw/RYJA/CMuRawLWKt/z1zPhzmvknT
-KfAIc6SjbBqu8Nx/Xvcd61c2V39U/nZDTs+H9QIDAQABAoIBAQC0gx8EfMgWBLbF
-WORKAaCRyKKPl8zWksCYPVAFLCnwLvf+VFRz7JJatofz/hMZn9K9Rd2EdhqELO42
-CMYhnneDOasRUzlPyMSgu1m4UezuYTopjX485Um/T2RGvdFrGw/qOKpZ+2i9XNzL
-c3Cf7KZHljERxirQqD+7hwGlMsxlCYpIRYbe6orpT0aAiGr1iVioohyk8tT+7iXD
-mlPeF9qbWAChgfLzTmHcpxGpiFXS0w6KV1XG8sickm1tnoXbCV5ZJw6HscL5VBp9
-SBclRo8AXWBaqhcfj0mvLJWs5E5K3P6dM9X/RcxJwP4Q+kfABYoYjZrZ1/sOJkZr
-mHzoYznRAoGBAPeDWWG5RLSlYgs6Llw9ERF3917AiY+eWUCai7faWGP3SEigmPOm
-m7rHQcI650DN22aXm8DBSRM1QV0C/UWd1DoDpQAc76GexJhLYRA8/ZGAe24/q1nX
-V0aDHzTLC6m8fYGj5ATOotvzYYz6aK9dCuLxPYfWmyZsIXU6K+ypsZOrAoGBAMKa
-a+7es33C8aUf6kvBOtPPs9tJW8rhox9gMQxHXuibz+0ZUM7h0DUbsMkDpj4Ydwqr
-c7O5sIHUjJLSmw7Oaw6xByK51tZNZeA3bVB2ZUAPILNkF2UqldRYRT7DfLTHEbNV
-DMo5P4tR8HxWdrKuhcp+RdPfVmay7iIkFtcKWbLfAoGAR34nKTUMhWln4npRvc7d
-yT/vsezHTzab7S82wEpPUcCxnljVFTvAq7i2Y9YDyhIsF3wfPxQVeXjegnFEmwE1
-tfQritbQ2Mw1WRAc30XesFJ+VKALbI3o5bMmJmen3MVXM0UVrdXJ8OJiAQiriEvF
-wzuPXFc+xWBiYawF1/xEELUCgYAYBm6K2A262gVxSGZpodp8aekfiof9nSvBZOPJ
-S0ppV0stT3HNiM1msRt7RasRgX244H/xUVx8Otx8B+pCwrMu5iYmYGEopfeM3eru
-Ax/u768u1o2Y3NAQnjE2VXYg7269ACQLF1REBAK3pwkSeD9mR36hcLI/DZoetuvm
-8o0uawKBgQCGwazVcS1cOOODw8mP39KIlWNtbpeRI6T4Qz8Z4FO5B324uccRmxL6
-gCRTdO0990mYQ/BJQ0EQWyrcnpqSfw1YYQOYpYvPexhKpRV+sU5KqJJkqEBe3WLm
-oiaFpuz+NtKkQrCc9AlA3SHlJWPo7jTjIPCpGuVM+FipGTbLGHr7HA==
+MIIEowIBAAKCAQEA4r+Hwd0sNf4wOlwefgQWG8SyU5hpxX2xwZ+Y6TRCRyTxJajd
+t1cdq8XZmMFgL7Mc/s44oBRcxkxGSiHz8JRa+xoBHFRok4Wg+EeWpn8EUk3RX2Qn
+7rovkPAuTdOcGOSv60RmrB9ZAI0bE0b353s67ptsfb8OXjlManINH8T2AMFXqxqP
+iYe0Fqfor9NYm8AGdh/G/NPkZJR4IdCW/KVvTEqYeDjKzbnRFGsaGSNZj7xWYK/E
+7kHxtWd0ipvs/pYuYFsbguUjsghUFh+0ZtjZ7VVB4A2b3m1eZsDlmSc83g9EbYDO
+ldAldH6y3INJ4CUKrhdCAhqAjAAmqcMxMAIWtwIDAQABAoIBAHJKCdpsaaFZWuO/
+lNr9oksizKOYGhAiN9uZXxCEd7e2ZhoOUJrdvSXclr4JEjtYs4+3UBFX0h9dSLyK
++pLTpDwWsmKeJQveefFK3CNVuE7nzFWhfQM2M02B0kUGGRLvCcCy/HFbZ5gAh5oi
+1keF9IFaM8f56fsSNHMew4C03S4pUqSQOXKuhXexJ5CfjN4oh+VqYRUK5MsLErXq
+hoDKc6h5Wwgv0fnSRuxjH7/4oaHUml7ydYm0aHFyed8gUhlbwdgeRTDCKhdjfDv0
+jJ7So9Cr84boya4crEWd733La73i+eE/IA/dDdKdMFmV9hXeutE2RISZhIyRKaVU
+M5aKjCkCgYEA/QIW7Fcy2aJkJUw68oEmLCQAHBbDgPB+vHr8lA4kHgpooptULQJt
+Eophk6uYNEFRbXcBuqFOFEF6BMR0ArUbJQjMqOmFZcjXYGWWuZPg5yd65ti7amiS
+UtfxY0zp2vdhDlcKKNzvzLRS1e9HSdtQIAsJO751q/t310CSCg5tBaUCgYEA5W3y
+MUpXNUfMk6DrNa1xDK/02oqV04YwhLCjOxWebhZYElLioZQuWfjp4ap2Q1VUIasT
+/EWrDeXrmPG3WOMlMKpkuF7eKH1u5DWztzI8MjPo+F/YSqfkvrVVVwJmbdlUanWX
+sLIt+qeknv3o2tAyQgREl9SKxjYUJ3bpDo74VCsCgYA+LYdIqd1bUuddqzBJbKfe
+7TIBtS5yv6fF1kvAbwE9SL2SSK4VKaV4iv2T9vW+0dabGkYYgY/YtoawKT+BctSq
+81CoqksYSpVzoUO7c7Yv2KoBlX3plq1vIpa2Gq7WGzmE9qsOg82C90XsQZO5eOZm
+GZA5WIM+stwu0DWTtxvf6QKBgBpjhQHqXRGByqQIVEV+R4yACJfCbO1Bd/8rGftH
+x0s+Qq8I3YxcMOCkYrJUGvovrFebB9dyY/RbR7f1xXLx3Ws1jhQuxTq/Aqw5CQy/
+OHFFyzwWOhiMKyFMVh6F5iYoYy9Di0kFlYhJFAQ9jDTu0XkKaKFWDwgxXCwV30SP
+aTy1AoGBAKhMm+BVwSmwKFyy6b5sBsK+xFejRfvWnvX0wa1nnGli9BvdnO4m4oXN
+/kk/RrR7FSP3B36flNOGQxv0FQUTYZMzUJywVgJo+tkviFLDRisugnfcaS7tfBkU
+20drAY+kM/bFoBZgtEGNAEvjFQ8Kpw29F0AWe5t++Dg7aNAhBXml
 -----END RSA PRIVATE KEY-----
Index: node-18.8.0/test/fixtures/keys/agent9-cert.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent9-cert.pem
+++ node-18.8.0/test/fixtures/keys/agent9-cert.pem
@@ -1,20 +1,20 @@
 -----BEGIN CERTIFICATE-----
-MIIDUjCCAjoCAQIwDQYJKoZIhvcNAQELBQAwfTELMAkGA1UEBhMCSUwxFjAUBgNV
+MIIDUDCCAjgCAQQwDQYJKoZIhvcNAQELBQAwfTELMAkGA1UEBhMCSUwxFjAUBgNV
 BAoMDVN0YXJ0Q29tIEx0ZC4xKzApBgNVBAsMIlNlY3VyZSBEaWdpdGFsIENlcnRp
 ZmljYXRlIFNpZ25pbmcxKTAnBgNVBAMMIFN0YXJ0Q29tIENlcnRpZmljYXRpb24g
-QXV0aG9yaXR5MCIYDzIwMTYxMDIxMDAwMDAxWhgPMjI5MjA4MzAxODQyMjFaMF0x
-CzAJBgNVBAYTAlVTMQswCQYDVQQIDAJDQTELMAkGA1UEBwwCU0YxDzANBgNVBAoM
-Bk5PREVKUzEPMA0GA1UECwwGYWdlbnQ5MRIwEAYDVQQDDAlsb2NhbGhvc3QwggEi
-MA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC2oEMk2EKwIZrx4IPNcGjHw5DO
-u8A8yJrcWG4pThUadrvwMI7bQ4QwNgHm4PVpbjAPbSUsRPX98PWL6GcpoH0lmJ9+
-j9CCEIEkW+j5wM7hYBXUSGuAZZfkdrpbZHsvwpYj2U39sfmUyGT1gBbGBmaAzODh
-ZaqYSm9VdaKS56SRMey3Pbsx+ikylgiEyPFoRKA141Zuxz1MKiwszLHuyz6pCZKY
-K7x1dlEGi3h3dvkRAdMyeSXJkYCZGbS5Fbl2OuW4pSWP4no/M960vBwEYvuJPDtx
-qxGezE51oXp4W4l9k+TYPOfGJDVW0PAg+JpfbepLetgFaO9/eNWes34AhF6FAgMB
-AAEwDQYJKoZIhvcNAQELBQADggEBAD8ojlI4FdXLCyHVYwwTP6BncF4tfeGP82/i
-Zcr8U9k28T2vlsLwdCGu8UVqGWfYrSY5oZqZmHPcDfZmv6Uc39eQ72aweoyLedk3
-UF1Ucwq+MxEM98doLlqL4lnPO1+TcpdhtoHAgT28WkddbR3alfsu+GRU3br3s4lS
-DHcm6UzdA/lkgZtC8wFUSW04WhzSHB78gm8VOl+1JGY0pp/T+ae5swkfj45Q3jOd
-H6jdZiUrU+LJQwLlXYniF4qzmH0SN8Gd3djVNzWJtNF+LFKXzCOYSK8AFaQ6Ta+s
-Pd6Rqa8Hl6cMmlsDu1NLumstvGna5wsc7ks1VZwtWt6WfIyIN2k=
+QXV0aG9yaXR5MCAXDTE2MTAyMTAwMDAwMVoYDzIyOTYwNjE3MTEwOTEzWjBdMQsw
+CQYDVQQGEwJVUzELMAkGA1UECAwCQ0ExCzAJBgNVBAcMAlNGMQ8wDQYDVQQKDAZO
+T0RFSlMxDzANBgNVBAsMBmFnZW50OTESMBAGA1UEAwwJbG9jYWxob3N0MIIBIjAN
+BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAv8DDbZ3WBYgR3c0F5BOGJz5OVhfG
+TVX9xCSB9RtM6rZpn45AcNv7wYlts4HZTmufspUvPvZPtBMlpZzW2W5wZ8L7vzsl
+HyM04tJavRaoYt/fLa/3yVeht0CeNc0JzgdE8JX7R7c8TmD384+Kc55aoKGOcZS6
+7mBXeJL6SX+CdlNw8+QuwHCAxxLR34F9AC2FPMlNXI6zNqKtuJZvHgccRQwNrmnz
+pVS8YHjLjEzqgQI1rVwDjAGMD8pOs/aDch2fj7wmoiu+/Yw9UvZIdlF3YuiPRtKQ
+3YrvkLTjpz0SuFSRegXHMkFKxd0YelqjkQOAAes6yqHUkKnnFjhb5SUmnQIDAQAB
+MA0GCSqGSIb3DQEBCwUAA4IBAQCWOdOjtiSsaTW5Xe0xI88gt9GfXwA3xZeiYjUl
+nuqcCRXtkE4H0CsqDgt/bH0rzKKmpKKVyaouSrDPsC1GNnmxmzavwh+wRVOEFSOf
+L4OBb9GLKxPbG6JluACAhzsJo2+TwTsJTZvB40PStrdIEw7BkwZ/rhMGnXQ3Asww
+jYVS7simdLCKK/r3W0tFh6KYUUOkJHU0ZF6YTkuBf7Mnu7ZM5DLZMQwfo9qg+Oe9
+e2Of6jgnffVAAmIMBVrdFd4YYBAk5bTTid3qWOX2oJH2H2Esa/ATIXnftfBeM9pg
+2fFUvautm1BoiyonLdL+rX1e6op5JVtuQ0gBk3Gml2jqlEPx
 -----END CERTIFICATE-----
Index: node-18.8.0/test/fixtures/keys/agent9-csr.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent9-csr.pem
+++ node-18.8.0/test/fixtures/keys/agent9-csr.pem
@@ -1,17 +1,17 @@
 -----BEGIN CERTIFICATE REQUEST-----
 MIICxzCCAa8CAQAwXTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAkNBMQswCQYDVQQH
 DAJTRjEPMA0GA1UECgwGTk9ERUpTMQ8wDQYDVQQLDAZhZ2VudDkxEjAQBgNVBAMM
-CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALagQyTY
-QrAhmvHgg81waMfDkM67wDzImtxYbilOFRp2u/AwjttDhDA2Aebg9WluMA9tJSxE
-9f3w9YvoZymgfSWYn36P0IIQgSRb6PnAzuFgFdRIa4Bll+R2ultkey/CliPZTf2x
-+ZTIZPWAFsYGZoDM4OFlqphKb1V1opLnpJEx7Lc9uzH6KTKWCITI8WhEoDXjVm7H
-PUwqLCzMse7LPqkJkpgrvHV2UQaLeHd2+REB0zJ5JcmRgJkZtLkVuXY65bilJY/i
-ej8z3rS8HARi+4k8O3GrEZ7MTnWhenhbiX2T5Ng858YkNVbQ8CD4ml9t6kt62AVo
-73941Z6zfgCEXoUCAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
-YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAms/rMyW2wVfNtBHZ7HAUxUlWl8ow
-0dlZgVmAXF0WBnfOGn31CQcGPyj9btJ48tmaTXmhIw96KqQDIi8KmYXDrDm0JmEp
-d+6Q704A0Qjwq4OmMqSobNHRVZUM24niF+U/oGuI8J5nSbCp/6m6chwM+R015cfl
-1yNqqQXYYIogcHQZVdofeKvGmrQhBfsEt5cdk2riGqfWVBwY6rfXW+MSHIw6cHIn
-vVFYG32Gk8ZU+MoWPQ/DLAy8B7Azo7ePMnidfaOxPAox6IGzCcZOfnCu2tZ09S5t
-gqcpdnecBLuQdIybhKhCbM7GOmIricDeIJXkVhmwmjpcu1WdQWUIUsD18A==
+CWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAL/Aw22d
+1gWIEd3NBeQThic+TlYXxk1V/cQkgfUbTOq2aZ+OQHDb+8GJbbOB2U5rn7KVLz72
+T7QTJaWc1tlucGfC+787JR8jNOLSWr0WqGLf3y2v98lXobdAnjXNCc4HRPCV+0e3
+PE5g9/OPinOeWqChjnGUuu5gV3iS+kl/gnZTcPPkLsBwgMcS0d+BfQAthTzJTVyO
+szairbiWbx4HHEUMDa5p86VUvGB4y4xM6oECNa1cA4wBjA/KTrP2g3Idn4+8JqIr
+vv2MPVL2SHZRd2Loj0bSkN2K75C046c9ErhUkXoFxzJBSsXdGHpao5EDgAHrOsqh
+1JCp5xY4W+UlJp0CAwEAAaAlMCMGCSqGSIb3DQEJBzEWDBRBIGNoYWxsZW5nZSBw
+YXNzd29yZDANBgkqhkiG9w0BAQsFAAOCAQEAQ8Ph5ugfDlKbAhugFjGAfA++wHn3
+H59haXoyicZeyBrS5GZ7Y4SkNrQSveFnLMpOLvYd5rKPZpKkuc1QAV6M7QbZYdum
+DWCY+l2OOM3BvMRV8OsCLRJ1iymjuo7r0uWK7zkpB9qiGShdJqWIUgcVVGDmneWa
+asmf0aIgNwhM+9PcggCBr859UltK/4lB1NxaDHOit1kG15ypJvWdrtoYeHbo4fn4
+FYP43Yhi6KEBI2m17Yj070NG2KQIT5g+A4n74CV0iEc9jCY9O6N91rXhZSBU2sK0
+ROBC7HcXgnK4FF8yRSWVXTMurItQcMmVt0NRZG3taHbt6AHMJ4OE/V7X3g==
 -----END CERTIFICATE REQUEST-----
Index: node-18.8.0/test/fixtures/keys/agent9-key.pem
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/agent9-key.pem
+++ node-18.8.0/test/fixtures/keys/agent9-key.pem
@@ -1,27 +1,27 @@
 -----BEGIN RSA PRIVATE KEY-----
-MIIEpQIBAAKCAQEAtqBDJNhCsCGa8eCDzXBox8OQzrvAPMia3FhuKU4VGna78DCO
-20OEMDYB5uD1aW4wD20lLET1/fD1i+hnKaB9JZiffo/QghCBJFvo+cDO4WAV1Ehr
-gGWX5Ha6W2R7L8KWI9lN/bH5lMhk9YAWxgZmgMzg4WWqmEpvVXWikuekkTHstz27
-MfopMpYIhMjxaESgNeNWbsc9TCosLMyx7ss+qQmSmCu8dXZRBot4d3b5EQHTMnkl
-yZGAmRm0uRW5djrluKUlj+J6PzPetLwcBGL7iTw7casRnsxOdaF6eFuJfZPk2Dzn
-xiQ1VtDwIPiaX23qS3rYBWjvf3jVnrN+AIRehQIDAQABAoIBAQCxMp0TkfZa+bBi
-woqAenJgaeQGg2vKTobcB72TvFyDmfNO4X6rRz5qnOyJfXsBelWNkkSASMU6SWOn
-Ba+bV0o2gXk4DwisOqFjiv5p3uedDGMB3+bW5TxVA9JcPQm91JtjW0TuRJK7BxnW
-jxsJt0ob7S7B5Kh7LbYLAKHm0nX+HgFcfiaRsVCiCtuYkTOzpkUQsl7psX4VJJQ+
-SYr4URsPZI7SGdIeq6ofKH/yGJeoYyhxfNDzNZXqdIC1LEU4BrsrBR6mVRTj35+K
-0pHiOQ5Eezi5BnEFUgRR6ompTXQGfnDgEmNyuhzVhuP6nFJ/QPO2KUcIkU4cqViC
-7ZTeaLjdAoGBAPNfZbi515DhtwdwDx/Ol0Y6kzyK2NP2FpMlGXwUKXSYbmwbSxSo
-zXGyCjxmKtQfbwzVSd/aNdH1lPCHerNFmumhL52wsWi/9EIT+liaxN8Cw/IbUuOy
-QlA8BT3Pf1H4Fv/ePIx1DuiPk5jp/eyAK/iPVCyXu7d85ULdfhvwVS7XAoGBAMAZ
-/x7dP7qRAUgjvCw8JQVpJPxeVvraKHpzQnceIhd88oTAIYHONNomSBAlqoT00JzW
-uFkDbtMZydhorOCUR/d6oHWyg4qEt7F89mssvCFXkROs+ePCuvspYO4F9B7/A7HT
-dwErWsvEaRmEO/AWRvHMhjTV4F5ZMNvPj3E1YX4DAoGBAMM/y7oRzrG7hD2BV4Dr
-G04KfElcE2ypx56xauqyujeCe0Rb+TZP3tLSRYgDZ2Ta+xrOmv/ubrNNVPpLltLw
-isHYwPy/3vTs2yeQI46mTD+mVlGMPknSn4UDQik+qSS35qvMPcNpvlYxqfZJ85+j
-jKNTSfKkoMMqfjvQuvXrMEvtAoGAIjZ/EWgmKXwZ1ldG9Dnh/gyz4Z6LrzGbc/OD
-KuPa/oPqTWpKjWvETfXzb6zFqdhQLx6uxmuuGTrGkBxUbcr65kCYw11/v/PTI3E2
-EfBtsSJ/XBm6h63uzzyXXs0ApWSVq94Vm8e07AWXEkxSwHe3OulKHa7ZvvPzl7Jn
-wanYKzECgYEAun2EZ1ETQ10fkQ9RYxeJYd3HjazTfWXUJJeVhPlsLQQyVfZoRwy0
-9nzdOclgSvE2/+m3SQFh4UHTddlrU1C/V8pkgImnOyCmBFEv8SAfuvEnhQWA13kA
-bfreLGMiDnbK1+7MEq0CJeIUZ1LWeuHlFNJT8bYvyjLiZcJyipgQ3NA=
+MIIEogIBAAKCAQEAv8DDbZ3WBYgR3c0F5BOGJz5OVhfGTVX9xCSB9RtM6rZpn45A
+cNv7wYlts4HZTmufspUvPvZPtBMlpZzW2W5wZ8L7vzslHyM04tJavRaoYt/fLa/3
+yVeht0CeNc0JzgdE8JX7R7c8TmD384+Kc55aoKGOcZS67mBXeJL6SX+CdlNw8+Qu
+wHCAxxLR34F9AC2FPMlNXI6zNqKtuJZvHgccRQwNrmnzpVS8YHjLjEzqgQI1rVwD
+jAGMD8pOs/aDch2fj7wmoiu+/Yw9UvZIdlF3YuiPRtKQ3YrvkLTjpz0SuFSRegXH
+MkFKxd0YelqjkQOAAes6yqHUkKnnFjhb5SUmnQIDAQABAoIBAFiH9TK89r1jD2yq
+Zhfhc8TWnXBeVa4LD7jpah7yVdbgFXAXS2VQC7JVcxK31s7AHeFjw4Z8TlxDotQN
+z5xfwwIFIhougnsX1CkNGdvHhELyPaZoYizRitaCvQj1+5rl/Exf0r5PuUYuuwv2
+r+Olrw4OySdjldLNJFWE9o3Mkz1xiDvYoHoiPaz87bw3Upuh124TfoAT/rM+dAd1
+GMbGiWRiUKp0Rsq5MNMYHRS6aAJr7UJ7iNp1HilvePgZ8ggHPo2dL/7PcpW5QkIk
+81afMVbzDK1cLDQSF+UCnYVJ2f7fx8eK1Rgb+0aEQTJ0BSD3InwvQV+d7k6iKQMW
+jdUs+PkCgYEA5G5x9bcA7i8BDnx08JAm7+o+rBVLbNZ5uZw6o98BsnJe/Ro76IR7
+B0N63R+xxVaIVjQ9WVy3mNQp6CxGIgKXu+CAaZvNWElOKv6j5JrTapJKwwT2Le59
+fOU8qV+XD4mT93nrfCwa1RuEs4q6QUo+mBXCcG2yqxVUX3towkgjA1sCgYEA1uUc
+bgtb2BzT3AxaYP4xgNTk6pWafy+0eXvnGo6BbVFd+74NPWArY8EEfsnP4dRmE/Ev
+fXH+BQvT0gie0LnqG8EAtCzV0Z5us//Bs6TqzXYAGJFzZPqswsPh7fdhQD6xp5gD
+/FDZFZ3lMpKeGmMXylVK6EFb0DfMQF+BiPcW92cCgYAmJi/QCnUUUjNJ+Q0Kk/h8
+vtEiqhfDoHoJvoaokXjWFXW5rWTt6xok12cwgLeOrmnsg6LvRqW+10f8Z1B7I5W1
+ocOwX33Gzc9JLOewKQ8/0Bi+QxjeCONaNWqRGs6+oIpenTTvsun0Rzk9RuDeQIj9
+mzyyzA83tUM3AwQ21hdK2wKBgCphgwrc6J8liE7kjMqGN0MeOYteN9sPkHtJCVn+
+ct4ZZvyoDbpmBgilX3qV2I7EMeGK9n+6BIwAOiU5GAE4OUiB9t6R3RV7Nn//V7dT
+Ng+RTxGJVgW7Pmkm8p7Y4/AkrSrNWzPeygFuY25m5hF0K5P24FAP28ijyH+Flmlp
+ubjPAoGANY8QRA8myAbrtOad6znmfr8Sa3v4dAFaNfVOWFhYyLtJXRb+KXLGm8nF
+nrxmRwD8/OR0mbODZ4b5JesoJKi5FWLfyMR2HKlR4Mg0LWPLbJ+214dy4ECN2N8m
+4wuQudJegScper89t8k4/zUgBHhosdCn6YcRUtMnw4TpZEpGwh8=
 -----END RSA PRIVATE KEY-----
