Index: node-18.8.0/test/fixtures/keys/Makefile
===================================================================
--- node-18.8.0.orig/test/fixtures/keys/Makefile
+++ node-18.8.0/test/fixtures/keys/Makefile
@@ -111,7 +111,7 @@ ca2-cert.pem: ca2.cnf
 # ('password' is used for the CA password.)
 #
 ca3-key.pem:
-	openssl genrsa -out ca3-key.pem 1024
+	openssl genrsa -out ca3-key.pem 4096
 
 ca3-csr.pem: ca3.cnf ca3-key.pem
 	openssl req -new \
@@ -137,7 +137,7 @@ ca3-cert.pem: ca3-csr.pem ca3-key.pem ca
 # ('password' is used for the CA password.)
 #
 ca4-key.pem:
-	openssl genrsa -out ca4-key.pem 1024
+	openssl genrsa -out ca4-key.pem 4096
 
 ca4-csr.pem: ca4.cnf ca4-key.pem
 	openssl req -new \
@@ -240,7 +240,7 @@ fake-startcom-root-cert.pem: fake-startc
 #
 
 agent1-key.pem:
-	openssl genrsa -out agent1-key.pem 1024
+	openssl genrsa -out agent1-key.pem 4096
 
 agent1-csr.pem: agent1.cnf agent1-key.pem
 	openssl req -new -config agent1.cnf -key agent1-key.pem -out agent1-csr.pem
@@ -275,7 +275,7 @@ agent1-verify: agent1-cert.pem ca1-cert.
 #
 # Generate new private key
 agent2-key.pem:
-	openssl genrsa -out agent2-key.pem 1024
+	openssl genrsa -out agent2-key.pem 4096
 
 # Create a Certificate Signing Request for the key
 agent2-csr.pem: agent2-key.pem agent2.cnf
@@ -297,7 +297,7 @@ agent2-verify: agent2-cert.pem
 #
 
 agent3-key.pem:
-	openssl genrsa -out agent3-key.pem 1024
+	openssl genrsa -out agent3-key.pem 4096
 
 agent3-csr.pem: agent3.cnf agent3-key.pem
 	openssl req -new -config agent3.cnf -key agent3-key.pem -out agent3-csr.pem
@@ -321,7 +321,7 @@ agent3-verify: agent3-cert.pem ca2-cert.
 #
 
 agent4-key.pem:
-	openssl genrsa -out agent4-key.pem 1024
+	openssl genrsa -out agent4-key.pem 4096
 
 agent4-csr.pem: agent4.cnf agent4-key.pem
 	openssl req -new -config agent4.cnf -key agent4-key.pem -out agent4-csr.pem
@@ -363,7 +363,7 @@ ca2-crl.pem: ca2-key.pem ca2-cert.pem ca
 #
 
 agent5-key.pem:
-	openssl genrsa -out agent5-key.pem 1024
+	openssl genrsa -out agent5-key.pem 4096
 
 agent5-csr.pem: agent5.cnf agent5-key.pem
 	openssl req -new -config agent5.cnf -key agent5-key.pem -out agent5-csr.pem
@@ -388,7 +388,7 @@ agent5-verify: agent5-cert.pem ca2-cert.
 #
 
 agent6-key.pem:
-	openssl genrsa -out agent6-key.pem 1024
+	openssl genrsa -out agent6-key.pem 4096
 
 agent6-csr.pem: agent6.cnf agent6-key.pem
 	openssl req -new -config agent6.cnf -key agent6-key.pem -out agent6-csr.pem
@@ -499,7 +499,7 @@ agent9-cert.pem: agent9-csr.pem
 #
 
 agent10-key.pem:
-	openssl genrsa -out agent10-key.pem 1024
+	openssl genrsa -out agent10-key.pem 4096
 
 agent10-csr.pem: agent10.cnf agent10-key.pem
 	openssl req -new -config agent10.cnf -key agent10-key.pem -out agent10-csr.pem
@@ -627,7 +627,7 @@ dsa_public_1025.pem:
 	openssl dsa -in dsa_private_1025.pem -pubout -out dsa_public_1025.pem
 
 rsa_private.pem:
-	openssl genrsa -out rsa_private.pem 2048
+	openssl genrsa -out rsa_private.pem 4096
 
 rsa_private_encrypted.pem: rsa_private.pem
 	openssl rsa -aes256 -in rsa_private.pem -passout 'pass:password' -out rsa_private_encrypted.pem
@@ -657,7 +657,7 @@ rsa_public_sha1_signature_signedby_rsa_p
 	openssl dgst -sha1 -sign rsa_private_pkcs8.pem -out rsa_public_sha1_signature_signedby_rsa_private_pkcs8.sha1 rsa_public.pem
 
 rsa_private_b.pem:
-	openssl genrsa -out rsa_private_b.pem 2048
+	openssl genrsa -out rsa_private_b.pem 4096
 
 I_AM_THE_WALRUS_sha256_signature_signedby_rsa_private_b.sha256: rsa_private_b.pem
 	echo -n "I AM THE WALRUS" | openssl dgst -sha256 -sign rsa_private_b.pem -out I_AM_THE_WALRUS_sha256_signature_signedby_rsa_private_b.sha256
